{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# BTE -- Question #2 -- Use Case Workflow"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 0. Uploads, Functions, and Parameters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import pandas and biothings explorers modules\n",
    "import pandas as pd\n",
    "from biothings_explorer.user_query_dispatcher import FindConnection\n",
    "from biothings_explorer.hint import Hint\n",
    "ht = Hint()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# set list constant that represents every node type available in BTE\n",
    "ALL_NODE_TYPES = ['Gene', 'SequenceVariant', 'ChemicalSubstance', 'Disease', \n",
    "                'MolecularActivity', 'BiologicalProcess', 'CellularComponent', \n",
    "                'Pathway', 'AnatomicalEntity', 'PhenotypicFeature']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# predict_many funciton will be used to run many BTE queries and return results as a single table\n",
    "def predict_many(input_object_list, output_type_list, intermediate_node_list = ''):\n",
    "    df_list = []\n",
    "    for input_object in input_object_list: \n",
    "        for output_type in output_type_list: \n",
    "            if(len(intermediate_node_list) > 0):\n",
    "                for inter in intermediate_node_list:\n",
    "                    try: \n",
    "                        print(\"Running: \" + input_object['name'] + ' --> intermediate type ' + inter + ' --> output type ' + output_type )\n",
    "                        fc = FindConnection(input_obj=input_object, output_obj=output_type, intermediate_nodes=[inter])\n",
    "                        fc.connect(verbose=False)\n",
    "                        df = fc.display_table_view()\n",
    "                        rows = df.shape[0]\n",
    "                        if(rows > 0):\n",
    "                            df_list.append(df)\n",
    "                    except:\n",
    "                        print(input_object['name'] + ' --> intermediate type ' + inter + ' --> output type ' + output_type + ' FAILED')\n",
    "            else:\n",
    "                try:\n",
    "                    print(\"Running: \" + input_object['name'] + ' --> output type ' + output_type )\n",
    "                    fc = FindConnection(input_obj=input_object, output_obj=output_type, intermediate_nodes=None)\n",
    "                    fc.connect(verbose=False)\n",
    "                    df = fc.display_table_view()\n",
    "                    rows = df.shape[0]\n",
    "                    if(rows > 0):\n",
    "                        df_list.append(df)\n",
    "                except:\n",
    "                    print(input_object['name'] + ' --> output type ' + output_type + ' FAILED')\n",
    "\n",
    "    if(len(df_list) > 0):\n",
    "        return pd.concat(df_list)\n",
    "    else:\n",
    "        return None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# max_one_step_genes represents the number of genes returned from direct disease -> gene query \n",
    "# that will be included in results. Genes with most occurrences will be included over those with less\n",
    "max_one_step_genes = 50\n",
    "\n",
    "# max_two_step_genes represents the number of genes returned from disease -> intermediate node -> gene query \n",
    "max_two_step_genes = 100\n",
    "\n",
    "# set disease name\n",
    "disease_name = 'thrombophilia due to activated protein C resistance'\n",
    "\n",
    "# set disease output files\n",
    "disease_csv_file = 'Dravet_Syndrome_BTE_2020_09_09.csv'\n",
    "disease_symptoms_csv = 'Dravet_Syndrome_Symptoms_2020_09_09.csv'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Get Disease Symptoms and Symptom Information "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.1 Get Disease"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'MONDO': 'MONDO:0008560', 'UMLS': 'C1861171', 'name': 'thrombophilia due to activated protein C resistance', 'MESH': 'D020016', 'OMIM': '188055', 'primary': {'identifier': 'MONDO', 'cls': 'Disease', 'value': 'MONDO:0008560'}, 'display': 'MONDO(MONDO:0008560) OMIM(188055) UMLS(C1861171) MESH(D020016) name(thrombophilia due to activated protein C resistance)', 'type': 'Disease'}\n"
     ]
    }
   ],
   "source": [
    "# run hint query to get disease input\n",
    "disease = ht.query(disease_name)['Disease'][0]\n",
    "print(disease)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.2 Get 'PhenotypicFeatures' Related to Disease"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>input</th>\n",
       "      <th>input_type</th>\n",
       "      <th>pred1</th>\n",
       "      <th>pred1_source</th>\n",
       "      <th>pred1_api</th>\n",
       "      <th>pred1_pubmed</th>\n",
       "      <th>output_type</th>\n",
       "      <th>output_name</th>\n",
       "      <th>output_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>None</td>\n",
       "      <td>BioLink API</td>\n",
       "      <td>None</td>\n",
       "      <td>PhenotypicFeature</td>\n",
       "      <td>BLOOD CLOT IN A DEEP VEIN</td>\n",
       "      <td>UMLS:C0149871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>hpo</td>\n",
       "      <td>mydisease.info API</td>\n",
       "      <td>None</td>\n",
       "      <td>PhenotypicFeature</td>\n",
       "      <td>BLOOD CLOT IN A DEEP VEIN</td>\n",
       "      <td>UMLS:C0149871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>None</td>\n",
       "      <td>BioLink API</td>\n",
       "      <td>None</td>\n",
       "      <td>PhenotypicFeature</td>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>UMLS:C0600433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>hpo</td>\n",
       "      <td>mydisease.info API</td>\n",
       "      <td>None</td>\n",
       "      <td>PhenotypicFeature</td>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>UMLS:C0600433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>None</td>\n",
       "      <td>BioLink API</td>\n",
       "      <td>None</td>\n",
       "      <td>PhenotypicFeature</td>\n",
       "      <td>ABNORMAL PARTIAL THROMBOPLASTIN TIME</td>\n",
       "      <td>UMLS:C0240671</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>hpo</td>\n",
       "      <td>mydisease.info API</td>\n",
       "      <td>None</td>\n",
       "      <td>PhenotypicFeature</td>\n",
       "      <td>ABNORMAL PARTIAL THROMBOPLASTIN TIME</td>\n",
       "      <td>UMLS:C0240671</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>None</td>\n",
       "      <td>BioLink API</td>\n",
       "      <td>None</td>\n",
       "      <td>PhenotypicFeature</td>\n",
       "      <td>BLOOD HYPERVISCOSITY</td>\n",
       "      <td>UMLS:C0398623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>hpo</td>\n",
       "      <td>mydisease.info API</td>\n",
       "      <td>None</td>\n",
       "      <td>PhenotypicFeature</td>\n",
       "      <td>BLOOD HYPERVISCOSITY</td>\n",
       "      <td>UMLS:C0398623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>None</td>\n",
       "      <td>BioLink API</td>\n",
       "      <td>None</td>\n",
       "      <td>PhenotypicFeature</td>\n",
       "      <td>PRE-ECLAMPSIA</td>\n",
       "      <td>UMLS:C0032914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>hpo</td>\n",
       "      <td>mydisease.info API</td>\n",
       "      <td>None</td>\n",
       "      <td>PhenotypicFeature</td>\n",
       "      <td>PRE-ECLAMPSIA</td>\n",
       "      <td>UMLS:C0032914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>hpo</td>\n",
       "      <td>mydisease.info API</td>\n",
       "      <td>None</td>\n",
       "      <td>PhenotypicFeature</td>\n",
       "      <td>AUTOSOMAL DOMINANT</td>\n",
       "      <td>UMLS:C0443147</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             input input_type       pred1 pred1_source  \\\n",
       "0   ACTIVATED PROTEIN C RESISTANCE    Disease  related_to         None   \n",
       "1   ACTIVATED PROTEIN C RESISTANCE    Disease  related_to          hpo   \n",
       "2   ACTIVATED PROTEIN C RESISTANCE    Disease  related_to         None   \n",
       "3   ACTIVATED PROTEIN C RESISTANCE    Disease  related_to          hpo   \n",
       "4   ACTIVATED PROTEIN C RESISTANCE    Disease  related_to         None   \n",
       "5   ACTIVATED PROTEIN C RESISTANCE    Disease  related_to          hpo   \n",
       "6   ACTIVATED PROTEIN C RESISTANCE    Disease  related_to         None   \n",
       "7   ACTIVATED PROTEIN C RESISTANCE    Disease  related_to          hpo   \n",
       "8   ACTIVATED PROTEIN C RESISTANCE    Disease  related_to         None   \n",
       "9   ACTIVATED PROTEIN C RESISTANCE    Disease  related_to          hpo   \n",
       "10  ACTIVATED PROTEIN C RESISTANCE    Disease  related_to          hpo   \n",
       "\n",
       "             pred1_api pred1_pubmed        output_type  \\\n",
       "0          BioLink API         None  PhenotypicFeature   \n",
       "1   mydisease.info API         None  PhenotypicFeature   \n",
       "2          BioLink API         None  PhenotypicFeature   \n",
       "3   mydisease.info API         None  PhenotypicFeature   \n",
       "4          BioLink API         None  PhenotypicFeature   \n",
       "5   mydisease.info API         None  PhenotypicFeature   \n",
       "6          BioLink API         None  PhenotypicFeature   \n",
       "7   mydisease.info API         None  PhenotypicFeature   \n",
       "8          BioLink API         None  PhenotypicFeature   \n",
       "9   mydisease.info API         None  PhenotypicFeature   \n",
       "10  mydisease.info API         None  PhenotypicFeature   \n",
       "\n",
       "                             output_name      output_id  \n",
       "0              BLOOD CLOT IN A DEEP VEIN  UMLS:C0149871  \n",
       "1              BLOOD CLOT IN A DEEP VEIN  UMLS:C0149871  \n",
       "2         ACTIVATED PROTEIN C RESISTANCE  UMLS:C0600433  \n",
       "3         ACTIVATED PROTEIN C RESISTANCE  UMLS:C0600433  \n",
       "4   ABNORMAL PARTIAL THROMBOPLASTIN TIME  UMLS:C0240671  \n",
       "5   ABNORMAL PARTIAL THROMBOPLASTIN TIME  UMLS:C0240671  \n",
       "6                   BLOOD HYPERVISCOSITY  UMLS:C0398623  \n",
       "7                   BLOOD HYPERVISCOSITY  UMLS:C0398623  \n",
       "8                          PRE-ECLAMPSIA  UMLS:C0032914  \n",
       "9                          PRE-ECLAMPSIA  UMLS:C0032914  \n",
       "10                    AUTOSOMAL DOMINANT  UMLS:C0443147  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get phenotypes (signs and symptoms) related to diesase\n",
    "fc = FindConnection(input_obj=disease, output_obj='PhenotypicFeature', intermediate_nodes=None)\n",
    "fc.connect(verbose=False)\n",
    "disease_to_phenotypicFeature = fc.display_table_view()\n",
    "disease_to_phenotypicFeature"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Note: all equivalent names for the disease input are as follows:\n",
      "ACTIVATED PROTEIN C RESISTANCE\n",
      "APC RESISTANCE\n",
      "PCCF DEFICIENCY\n",
      "PROC COFACTOR DEFICIENCY\n",
      "RESISTANCE, APC\n",
      "THPH2\n",
      "THROMBOPHILIA 5\n",
      "THROMBOPHILIA DUE TO ACTIVATED PROTEIN C RESISTANCE\n",
      "THROMBOPHILIA DUE TO ACTIVATED PROTEIN C RESISTANCE; THPH2\n",
      "THROMBOPHILIA DUE TO DEFICIENCY OF ACTIVATED PROTEIN C COFACTOR\n",
      "THROMBOPHILIA DUE TO FACTOR 5 LEIDEN\n"
     ]
    }
   ],
   "source": [
    "# Print equivalent names for the disease input\n",
    "print('Note: all equivalent names for the disease input are as follows:')\n",
    "for name in fc.fc.display_node_info(disease_name)['equivalent_ids']['name']: print(name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'HP:0002625': {'names': ['BLOOD CLOT IN A DEEP VEIN', 'DEEP VEIN THROMBOSIS', 'DEEP VENOUS THROMBOSIS', 'MULTIPLE DEEP VENOUS THROMBOSIS'], 'frequency': 'Unknown'}, 'HP:0012175': {'names': ['ACTIVATED PROTEIN C RESISTANCE', 'RESISTANCE TO ACTIVATED PROTEIN C'], 'frequency': 'Unknown'}, 'HP:0003645': {'names': ['ABNORMAL PARTIAL THROMBOPLASTIN TIME', 'DELAYED THROMBOPLASTIN GENERATION', 'PARTIAL THROMBOPLASTIN TIME PROLONGED', 'PROLONGED ACTIVATED PARTIAL THROMBOPLASTIN TIME', 'PROLONGED PARTIAL THROMBOPLASTIN TIME', 'PROLONGED PTT'], 'frequency': 'Unknown'}, 'HP:0100724': {'names': ['BLOOD HYPERVISCOSITY', 'HYPERCOAGULABILITY', 'THROMBOPHILIA'], 'frequency': 'Unknown'}, 'HP:0100602': {'names': ['PRE-ECLAMPSIA', 'PREECLAMPSIA'], 'frequency': 'Unknown'}, 'HP:0000006': {'names': ['AUTOSOMAL DOMINANT', 'AUTOSOMAL DOMINANT FORM', 'AUTOSOMAL DOMINANT INHERITANCE', 'AUTOSOMAL DOMINANT TYPE'], 'frequency': 'Unknown'}}\n"
     ]
    }
   ],
   "source": [
    "# create dictionary of symptom HPIDs, nad symptom names (with synonyms)\n",
    "symptom_dict = {}\n",
    "for index, row in disease_to_phenotypicFeature.iterrows():\n",
    "    output_name = disease_to_phenotypicFeature['output_name'][index]\n",
    "    items = fc.fc.G[disease_name][output_name].values()\n",
    "    for item in items: \n",
    "#         print(item)\n",
    "        if('frequency' in item['info']):\n",
    "            freq = [_item['info']['frequency'] for _item in fc.fc.G[disease_name][output_name].values() if \"frequency\" in _item[\"info\"]][0][0]\n",
    "            freq_value = ht.query(freq)['PhenotypicFeature'][0]['name']\n",
    "        else: \n",
    "            freq_value = 'Unknown'\n",
    "                    \n",
    "        symptom_dict[fc.fc.display_node_info(output_name)['equivalent_ids']['HP'][0]] = {\n",
    "            \"names\": fc.fc.display_node_info(output_name)['equivalent_ids']['name'],\n",
    "            \"frequency\": freq_value,\n",
    "        }\n",
    "print(symptom_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create list of HPIDs and Symptoms for later use\n",
    "disease_symptom_hpids = list(symptom_dict.keys())\n",
    "disease_symptoms = []\n",
    "for entry in list(symptom_dict.values()): disease_symptoms = disease_symptoms + entry['names']\n",
    "\n",
    "\n",
    "# create dict of HPIDs : symptoms for use in assembling final results\n",
    "symptom_to_hpid_dict = {}\n",
    "for key,value in symptom_dict.items(): \n",
    "    for name in value['names']:\n",
    "        symptom_to_hpid_dict[name] = key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running: Deep venous thrombosis --> output type Gene\n",
      "Running: Deep venous thrombosis --> output type SequenceVariant\n",
      "Running: Deep venous thrombosis --> output type ChemicalSubstance\n",
      "Running: Deep venous thrombosis --> output type Disease\n",
      "Running: Deep venous thrombosis --> output type MolecularActivity\n",
      "Running: Deep venous thrombosis --> output type BiologicalProcess\n",
      "Running: Deep venous thrombosis --> output type CellularComponent\n",
      "Running: Deep venous thrombosis --> output type Pathway\n",
      "Running: Deep venous thrombosis --> output type AnatomicalEntity\n",
      "Running: Deep venous thrombosis --> output type PhenotypicFeature\n",
      "Running: Resistance to activated protein C --> output type Gene\n",
      "Running: Resistance to activated protein C --> output type SequenceVariant\n",
      "Running: Resistance to activated protein C --> output type ChemicalSubstance\n",
      "Running: Resistance to activated protein C --> output type Disease\n",
      "Running: Resistance to activated protein C --> output type MolecularActivity\n",
      "Running: Resistance to activated protein C --> output type BiologicalProcess\n",
      "Running: Resistance to activated protein C --> output type CellularComponent\n",
      "Running: Resistance to activated protein C --> output type Pathway\n",
      "Running: Resistance to activated protein C --> output type AnatomicalEntity\n",
      "Running: Resistance to activated protein C --> output type PhenotypicFeature\n",
      "Running: Prolonged partial thromboplastin time --> output type Gene\n",
      "Running: Prolonged partial thromboplastin time --> output type SequenceVariant\n",
      "Running: Prolonged partial thromboplastin time --> output type ChemicalSubstance\n",
      "Running: Prolonged partial thromboplastin time --> output type Disease\n",
      "Running: Prolonged partial thromboplastin time --> output type MolecularActivity\n",
      "Running: Prolonged partial thromboplastin time --> output type BiologicalProcess\n",
      "Running: Prolonged partial thromboplastin time --> output type CellularComponent\n",
      "Running: Prolonged partial thromboplastin time --> output type Pathway\n",
      "Running: Prolonged partial thromboplastin time --> output type AnatomicalEntity\n",
      "Running: Prolonged partial thromboplastin time --> output type PhenotypicFeature\n",
      "Running: Hypercoagulability --> output type Gene\n",
      "Running: Hypercoagulability --> output type SequenceVariant\n",
      "Running: Hypercoagulability --> output type ChemicalSubstance\n",
      "Running: Hypercoagulability --> output type Disease\n",
      "Running: Hypercoagulability --> output type MolecularActivity\n",
      "Running: Hypercoagulability --> output type BiologicalProcess\n",
      "Running: Hypercoagulability --> output type CellularComponent\n",
      "Running: Hypercoagulability --> output type Pathway\n",
      "Running: Hypercoagulability --> output type AnatomicalEntity\n",
      "Running: Hypercoagulability --> output type PhenotypicFeature\n",
      "Running: Preeclampsia --> output type Gene\n",
      "Running: Preeclampsia --> output type SequenceVariant\n",
      "Running: Preeclampsia --> output type ChemicalSubstance\n",
      "Running: Preeclampsia --> output type Disease\n",
      "Running: Preeclampsia --> output type MolecularActivity\n",
      "Running: Preeclampsia --> output type BiologicalProcess\n",
      "Running: Preeclampsia --> output type CellularComponent\n",
      "Running: Preeclampsia --> output type Pathway\n",
      "Running: Preeclampsia --> output type AnatomicalEntity\n",
      "Running: Preeclampsia --> output type PhenotypicFeature\n",
      "Running: Autosomal dominant inheritance --> output type Gene\n",
      "Running: Autosomal dominant inheritance --> output type SequenceVariant\n",
      "Running: Autosomal dominant inheritance --> output type ChemicalSubstance\n",
      "Running: Autosomal dominant inheritance --> output type Disease\n",
      "Running: Autosomal dominant inheritance --> output type MolecularActivity\n",
      "Running: Autosomal dominant inheritance --> output type BiologicalProcess\n",
      "Running: Autosomal dominant inheritance --> output type CellularComponent\n",
      "Running: Autosomal dominant inheritance --> output type Pathway\n",
      "Running: Autosomal dominant inheritance --> output type AnatomicalEntity\n",
      "Running: Autosomal dominant inheritance --> output type PhenotypicFeature\n",
      "{'HP:0002625': {'names': ['BLOOD CLOT IN A DEEP VEIN', 'DEEP VEIN THROMBOSIS', 'DEEP VENOUS THROMBOSIS', 'MULTIPLE DEEP VENOUS THROMBOSIS'], 'frequency': 'Unknown', 'edges_out_count': 38}, 'HP:0012175': {'names': ['ACTIVATED PROTEIN C RESISTANCE', 'RESISTANCE TO ACTIVATED PROTEIN C'], 'frequency': 'Unknown', 'edges_out_count': 4}, 'HP:0003645': {'names': ['ABNORMAL PARTIAL THROMBOPLASTIN TIME', 'DELAYED THROMBOPLASTIN GENERATION', 'PARTIAL THROMBOPLASTIN TIME PROLONGED', 'PROLONGED ACTIVATED PARTIAL THROMBOPLASTIN TIME', 'PROLONGED PARTIAL THROMBOPLASTIN TIME', 'PROLONGED PTT'], 'frequency': 'Unknown', 'edges_out_count': 76}, 'HP:0100724': {'names': ['BLOOD HYPERVISCOSITY', 'HYPERCOAGULABILITY', 'THROMBOPHILIA'], 'frequency': 'Unknown', 'edges_out_count': 47}, 'HP:0100602': {'names': ['PRE-ECLAMPSIA', 'PREECLAMPSIA'], 'frequency': 'Unknown', 'edges_out_count': 36}, 'HP:0000006': {'names': ['AUTOSOMAL DOMINANT', 'AUTOSOMAL DOMINANT FORM', 'AUTOSOMAL DOMINANT INHERITANCE', 'AUTOSOMAL DOMINANT TYPE'], 'frequency': 'Unknown', 'edges_out_count': 538}}\n"
     ]
    }
   ],
   "source": [
    "# add \"edges out\" counts from each phenotype to any node type, to get a rough estimate of how prevalent a phenotype is\n",
    "phenotype_inputs = []\n",
    "for hpid in disease_symptom_hpids: \n",
    "    try: \n",
    "        phenotype_input = ht.query(hpid)['PhenotypicFeature'][0]\n",
    "        all_edges_out_df = predict_many([phenotype_input], ALL_NODE_TYPES)\n",
    "        symptom_dict[hpid]['edges_out_count'] = all_edges_out_df.shape[0]\n",
    "    except: \n",
    "        print(hpid + ' Failed')\n",
    "\n",
    "print(symptom_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>names</th>\n",
       "      <th>frequency</th>\n",
       "      <th>edges_out_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HP:0012175</th>\n",
       "      <td>[ACTIVATED PROTEIN C RESISTANCE, RESISTANCE TO...</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP:0100602</th>\n",
       "      <td>[PRE-ECLAMPSIA, PREECLAMPSIA]</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP:0002625</th>\n",
       "      <td>[BLOOD CLOT IN A DEEP VEIN, DEEP VEIN THROMBOS...</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP:0100724</th>\n",
       "      <td>[BLOOD HYPERVISCOSITY, HYPERCOAGULABILITY, THR...</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP:0003645</th>\n",
       "      <td>[ABNORMAL PARTIAL THROMBOPLASTIN TIME, DELAYED...</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HP:0000006</th>\n",
       "      <td>[AUTOSOMAL DOMINANT, AUTOSOMAL DOMINANT FORM, ...</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>538</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                        names frequency  \\\n",
       "HP:0012175  [ACTIVATED PROTEIN C RESISTANCE, RESISTANCE TO...   Unknown   \n",
       "HP:0100602                      [PRE-ECLAMPSIA, PREECLAMPSIA]   Unknown   \n",
       "HP:0002625  [BLOOD CLOT IN A DEEP VEIN, DEEP VEIN THROMBOS...   Unknown   \n",
       "HP:0100724  [BLOOD HYPERVISCOSITY, HYPERCOAGULABILITY, THR...   Unknown   \n",
       "HP:0003645  [ABNORMAL PARTIAL THROMBOPLASTIN TIME, DELAYED...   Unknown   \n",
       "HP:0000006  [AUTOSOMAL DOMINANT, AUTOSOMAL DOMINANT FORM, ...   Unknown   \n",
       "\n",
       "            edges_out_count  \n",
       "HP:0012175                4  \n",
       "HP:0100602               36  \n",
       "HP:0002625               38  \n",
       "HP:0100724               47  \n",
       "HP:0003645               76  \n",
       "HP:0000006              538  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# convert symptom dictionary to dataframe and sort by edges out and frequency \n",
    "disease_symptom_df = pd.DataFrame.from_dict(symptom_dict, orient='index').sort_values(by=['edges_out_count'])\n",
    "disease_symptom_df = pd.concat([disease_symptom_df[disease_symptom_df[\"frequency\"] == \"Very frequent\"],\n",
    "               disease_symptom_df[disease_symptom_df[\"frequency\"] == \"Frequent\"],\n",
    "               disease_symptom_df[disease_symptom_df[\"frequency\"] == \"Occasional\"],\n",
    "               disease_symptom_df[disease_symptom_df[\"frequency\"] == \"Rare\"],\n",
    "               disease_symptom_df[disease_symptom_df[\"frequency\"] == \"Unknown\"]\n",
    "              ])\n",
    "disease_symptom_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "disease_symptom_df.to_csv(disease_symptoms_csv, index = True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Get Genes Directly Related to Disease"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>input</th>\n",
       "      <th>input_type</th>\n",
       "      <th>pred1</th>\n",
       "      <th>pred1_source</th>\n",
       "      <th>pred1_api</th>\n",
       "      <th>pred1_pubmed</th>\n",
       "      <th>output_type</th>\n",
       "      <th>output_name</th>\n",
       "      <th>output_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>disrupted_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>7745572</td>\n",
       "      <td>Gene</td>\n",
       "      <td>CDSN</td>\n",
       "      <td>NCBIGene:1041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>affected_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>11703344,16493484</td>\n",
       "      <td>Gene</td>\n",
       "      <td>CDSN</td>\n",
       "      <td>NCBIGene:1041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>8834124</td>\n",
       "      <td>Gene</td>\n",
       "      <td>CDSN</td>\n",
       "      <td>NCBIGene:1041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>caused_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>19036061,19764414,19803829</td>\n",
       "      <td>Gene</td>\n",
       "      <td>CDSN</td>\n",
       "      <td>NCBIGene:1041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>disrupted_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>12624624</td>\n",
       "      <td>Gene</td>\n",
       "      <td>CD82</td>\n",
       "      <td>NCBIGene:3732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>affected_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>11703344</td>\n",
       "      <td>Gene</td>\n",
       "      <td>TFPI</td>\n",
       "      <td>NCBIGene:7035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>disgenet</td>\n",
       "      <td>mydisease.info API</td>\n",
       "      <td>None</td>\n",
       "      <td>Gene</td>\n",
       "      <td>TFPI</td>\n",
       "      <td>NCBIGene:7035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>pharos</td>\n",
       "      <td>Automat PHAROS API</td>\n",
       "      <td>None</td>\n",
       "      <td>Gene</td>\n",
       "      <td>TFPI</td>\n",
       "      <td>NCBIGene:7035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>7631299</td>\n",
       "      <td>Gene</td>\n",
       "      <td>KLK3</td>\n",
       "      <td>NCBIGene:354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>20161734</td>\n",
       "      <td>Gene</td>\n",
       "      <td>SERPINE1</td>\n",
       "      <td>NCBIGene:5054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>20161734</td>\n",
       "      <td>Gene</td>\n",
       "      <td>SERPINB2</td>\n",
       "      <td>NCBIGene:5055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>21846908</td>\n",
       "      <td>Gene</td>\n",
       "      <td>SSB</td>\n",
       "      <td>NCBIGene:6741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>caused_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>16315641,21846908</td>\n",
       "      <td>Gene</td>\n",
       "      <td>SSB</td>\n",
       "      <td>NCBIGene:6741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>18768782</td>\n",
       "      <td>Gene</td>\n",
       "      <td>JAK2</td>\n",
       "      <td>NCBIGene:3717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>21914670</td>\n",
       "      <td>Gene</td>\n",
       "      <td>TWIST1</td>\n",
       "      <td>NCBIGene:7291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>disgenet</td>\n",
       "      <td>mydisease.info API</td>\n",
       "      <td>None</td>\n",
       "      <td>Gene</td>\n",
       "      <td>F5</td>\n",
       "      <td>NCBIGene:2153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>scibite</td>\n",
       "      <td>Automat CORD19 Scibite API</td>\n",
       "      <td>None</td>\n",
       "      <td>Gene</td>\n",
       "      <td>F5</td>\n",
       "      <td>NCBIGene:2153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>None</td>\n",
       "      <td>BioLink API</td>\n",
       "      <td>None</td>\n",
       "      <td>Gene</td>\n",
       "      <td>F5</td>\n",
       "      <td>NCBIGene:2153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>disgenet</td>\n",
       "      <td>mydisease.info API</td>\n",
       "      <td>None</td>\n",
       "      <td>Gene</td>\n",
       "      <td>PROS1</td>\n",
       "      <td>NCBIGene:5627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>pharos</td>\n",
       "      <td>Automat PHAROS API</td>\n",
       "      <td>None</td>\n",
       "      <td>Gene</td>\n",
       "      <td>PROS1</td>\n",
       "      <td>NCBIGene:5627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>caused_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>16315641</td>\n",
       "      <td>Gene</td>\n",
       "      <td>TUBB4B</td>\n",
       "      <td>NCBIGene:10383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>caused_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>16315641</td>\n",
       "      <td>Gene</td>\n",
       "      <td>NEUROD1</td>\n",
       "      <td>NCBIGene:4760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>caused_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>9493610</td>\n",
       "      <td>Gene</td>\n",
       "      <td>CP</td>\n",
       "      <td>NCBIGene:1356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>caused_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>15637132</td>\n",
       "      <td>Gene</td>\n",
       "      <td>GPI</td>\n",
       "      <td>NCBIGene:2821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>scibite</td>\n",
       "      <td>Automat CORD19 Scibite API</td>\n",
       "      <td>None</td>\n",
       "      <td>Gene</td>\n",
       "      <td>F8</td>\n",
       "      <td>NCBIGene:2157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>scibite</td>\n",
       "      <td>Automat CORD19 Scibite API</td>\n",
       "      <td>None</td>\n",
       "      <td>Gene</td>\n",
       "      <td>PLAT</td>\n",
       "      <td>NCBIGene:5327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>scibite</td>\n",
       "      <td>Automat CORD19 Scibite API</td>\n",
       "      <td>None</td>\n",
       "      <td>Gene</td>\n",
       "      <td>PLG</td>\n",
       "      <td>NCBIGene:5340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>scibite</td>\n",
       "      <td>Automat CORD19 Scibite API</td>\n",
       "      <td>None</td>\n",
       "      <td>Gene</td>\n",
       "      <td>PROC</td>\n",
       "      <td>NCBIGene:5624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>scibite</td>\n",
       "      <td>Automat CORD19 Scibite API</td>\n",
       "      <td>None</td>\n",
       "      <td>Gene</td>\n",
       "      <td>LPA</td>\n",
       "      <td>NCBIGene:4018</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             input input_type         pred1 pred1_source  \\\n",
       "20  ACTIVATED PROTEIN C RESISTANCE    Disease  disrupted_by       SEMMED   \n",
       "21  ACTIVATED PROTEIN C RESISTANCE    Disease   affected_by       SEMMED   \n",
       "22  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to       SEMMED   \n",
       "23  ACTIVATED PROTEIN C RESISTANCE    Disease     caused_by       SEMMED   \n",
       "24  ACTIVATED PROTEIN C RESISTANCE    Disease  disrupted_by       SEMMED   \n",
       "25  ACTIVATED PROTEIN C RESISTANCE    Disease   affected_by       SEMMED   \n",
       "26  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to     disgenet   \n",
       "27  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to       pharos   \n",
       "29  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to       SEMMED   \n",
       "30  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to       SEMMED   \n",
       "31  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to       SEMMED   \n",
       "32  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to       SEMMED   \n",
       "33  ACTIVATED PROTEIN C RESISTANCE    Disease     caused_by       SEMMED   \n",
       "34  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to       SEMMED   \n",
       "35  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to       SEMMED   \n",
       "36  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to     disgenet   \n",
       "37  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to      scibite   \n",
       "39  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to         None   \n",
       "40  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to     disgenet   \n",
       "41  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to       pharos   \n",
       "43  ACTIVATED PROTEIN C RESISTANCE    Disease     caused_by       SEMMED   \n",
       "44  ACTIVATED PROTEIN C RESISTANCE    Disease     caused_by       SEMMED   \n",
       "45  ACTIVATED PROTEIN C RESISTANCE    Disease     caused_by       SEMMED   \n",
       "46  ACTIVATED PROTEIN C RESISTANCE    Disease     caused_by       SEMMED   \n",
       "47  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to      scibite   \n",
       "48  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to      scibite   \n",
       "49  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to      scibite   \n",
       "50  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to      scibite   \n",
       "51  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to      scibite   \n",
       "\n",
       "                     pred1_api                pred1_pubmed output_type  \\\n",
       "20          SEMMED Disease API                     7745572        Gene   \n",
       "21          SEMMED Disease API           11703344,16493484        Gene   \n",
       "22          SEMMED Disease API                     8834124        Gene   \n",
       "23          SEMMED Disease API  19036061,19764414,19803829        Gene   \n",
       "24          SEMMED Disease API                    12624624        Gene   \n",
       "25          SEMMED Disease API                    11703344        Gene   \n",
       "26          mydisease.info API                        None        Gene   \n",
       "27          Automat PHAROS API                        None        Gene   \n",
       "29          SEMMED Disease API                     7631299        Gene   \n",
       "30          SEMMED Disease API                    20161734        Gene   \n",
       "31          SEMMED Disease API                    20161734        Gene   \n",
       "32          SEMMED Disease API                    21846908        Gene   \n",
       "33          SEMMED Disease API           16315641,21846908        Gene   \n",
       "34          SEMMED Disease API                    18768782        Gene   \n",
       "35          SEMMED Disease API                    21914670        Gene   \n",
       "36          mydisease.info API                        None        Gene   \n",
       "37  Automat CORD19 Scibite API                        None        Gene   \n",
       "39                 BioLink API                        None        Gene   \n",
       "40          mydisease.info API                        None        Gene   \n",
       "41          Automat PHAROS API                        None        Gene   \n",
       "43          SEMMED Disease API                    16315641        Gene   \n",
       "44          SEMMED Disease API                    16315641        Gene   \n",
       "45          SEMMED Disease API                     9493610        Gene   \n",
       "46          SEMMED Disease API                    15637132        Gene   \n",
       "47  Automat CORD19 Scibite API                        None        Gene   \n",
       "48  Automat CORD19 Scibite API                        None        Gene   \n",
       "49  Automat CORD19 Scibite API                        None        Gene   \n",
       "50  Automat CORD19 Scibite API                        None        Gene   \n",
       "51  Automat CORD19 Scibite API                        None        Gene   \n",
       "\n",
       "   output_name       output_id  \n",
       "20        CDSN   NCBIGene:1041  \n",
       "21        CDSN   NCBIGene:1041  \n",
       "22        CDSN   NCBIGene:1041  \n",
       "23        CDSN   NCBIGene:1041  \n",
       "24        CD82   NCBIGene:3732  \n",
       "25        TFPI   NCBIGene:7035  \n",
       "26        TFPI   NCBIGene:7035  \n",
       "27        TFPI   NCBIGene:7035  \n",
       "29        KLK3    NCBIGene:354  \n",
       "30    SERPINE1   NCBIGene:5054  \n",
       "31    SERPINB2   NCBIGene:5055  \n",
       "32         SSB   NCBIGene:6741  \n",
       "33         SSB   NCBIGene:6741  \n",
       "34        JAK2   NCBIGene:3717  \n",
       "35      TWIST1   NCBIGene:7291  \n",
       "36          F5   NCBIGene:2153  \n",
       "37          F5   NCBIGene:2153  \n",
       "39          F5   NCBIGene:2153  \n",
       "40       PROS1   NCBIGene:5627  \n",
       "41       PROS1   NCBIGene:5627  \n",
       "43      TUBB4B  NCBIGene:10383  \n",
       "44     NEUROD1   NCBIGene:4760  \n",
       "45          CP   NCBIGene:1356  \n",
       "46         GPI   NCBIGene:2821  \n",
       "47          F8   NCBIGene:2157  \n",
       "48        PLAT   NCBIGene:5327  \n",
       "49         PLG   NCBIGene:5340  \n",
       "50        PROC   NCBIGene:5624  \n",
       "51         LPA   NCBIGene:4018  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# find genes directly related to disease\n",
    "fc = FindConnection(input_obj=disease, output_obj='Gene', intermediate_nodes=None)\n",
    "fc.connect(verbose=False)\n",
    "disease_to_genes = fc.display_table_view()\n",
    "disease_to_genes = disease_to_genes[~disease_to_genes['output_id'].str.contains('UMLS')]\n",
    "disease_to_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'CDSN': {'gene_count': 4,\n",
       "  'publications': ['7745572',\n",
       "   '11703344',\n",
       "   '16493484',\n",
       "   '8834124',\n",
       "   '19036061',\n",
       "   '19764414',\n",
       "   '19803829']},\n",
       " 'TFPI': {'gene_count': 3, 'publications': ['11703344']},\n",
       " 'F5': {'gene_count': 3, 'publications': []},\n",
       " 'SSB': {'gene_count': 2, 'publications': ['21846908', '16315641']},\n",
       " 'PROS1': {'gene_count': 2, 'publications': []},\n",
       " 'CD82': {'gene_count': 1, 'publications': ['12624624']},\n",
       " 'KLK3': {'gene_count': 1, 'publications': ['7631299']},\n",
       " 'SERPINE1': {'gene_count': 1, 'publications': ['20161734']},\n",
       " 'SERPINB2': {'gene_count': 1, 'publications': ['20161734']},\n",
       " 'JAK2': {'gene_count': 1, 'publications': ['18768782']},\n",
       " 'TWIST1': {'gene_count': 1, 'publications': ['21914670']},\n",
       " 'TUBB4B': {'gene_count': 1, 'publications': ['16315641']},\n",
       " 'NEUROD1': {'gene_count': 1, 'publications': ['16315641']},\n",
       " 'CP': {'gene_count': 1, 'publications': ['9493610']},\n",
       " 'GPI': {'gene_count': 1, 'publications': ['15637132']},\n",
       " 'F8': {'gene_count': 1, 'publications': []},\n",
       " 'PLAT': {'gene_count': 1, 'publications': []},\n",
       " 'PLG': {'gene_count': 1, 'publications': []},\n",
       " 'PROC': {'gene_count': 1, 'publications': []},\n",
       " 'LPA': {'gene_count': 1, 'publications': []}}"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create dictionary of gene reults whereby each Gene has values for the number of occurrences of the gene in results\n",
    "# and the number of publications connecting the disease -> gene directly\n",
    "disease_to_gene_results = {}\n",
    "disease_to_gene_genes = list(disease_to_genes[\"output_name\"]) # create list of genes\n",
    "disease_to_gene_genes = list(dict.fromkeys(disease_to_gene_genes))  # remove duplicates\n",
    "\n",
    "for gene in disease_to_gene_genes: \n",
    "    disease_to_gene_results[gene] = {\n",
    "        'gene_count' : 0,\n",
    "        'publications': []\n",
    "    }\n",
    "\n",
    "for index, row in disease_to_genes.iterrows():\n",
    "    disease_to_gene_results[row['output_name']]['gene_count'] = disease_to_gene_results[row['output_name']]['gene_count'] + 1\n",
    "    if(row['pred1_pubmed'] != None): \n",
    "        disease_to_gene_results[row['output_name']]['publications'] = disease_to_gene_results[row['output_name']]['publications'] + row['pred1_pubmed'].split(\",\")\n",
    "\n",
    "disease_to_gene_results = dict(sorted(disease_to_gene_results.items(), key = lambda x: x[1]['gene_count'], reverse = True))\n",
    "\n",
    "for key,value in disease_to_gene_results.items(): \n",
    "    disease_to_gene_results[key]['publications'] = list(dict.fromkeys(disease_to_gene_results[key]['publications']))\n",
    "    \n",
    "    \n",
    "disease_to_gene_results\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Get Top Genes Related to Disease through 1 Intermediate Node"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running: thrombophilia due to activated protein C resistance --> intermediate type Gene --> output type Gene\n",
      "Running: thrombophilia due to activated protein C resistance --> intermediate type SequenceVariant --> output type Gene\n",
      "Running: thrombophilia due to activated protein C resistance --> intermediate type ChemicalSubstance --> output type Gene\n",
      "API 6.1 pharos failed\n",
      "Running: thrombophilia due to activated protein C resistance --> intermediate type Disease --> output type Gene\n",
      "Running: thrombophilia due to activated protein C resistance --> intermediate type MolecularActivity --> output type Gene\n",
      "Running: thrombophilia due to activated protein C resistance --> intermediate type BiologicalProcess --> output type Gene\n",
      "Running: thrombophilia due to activated protein C resistance --> intermediate type CellularComponent --> output type Gene\n",
      "Running: thrombophilia due to activated protein C resistance --> intermediate type Pathway --> output type Gene\n",
      "Running: thrombophilia due to activated protein C resistance --> intermediate type AnatomicalEntity --> output type Gene\n",
      "Running: thrombophilia due to activated protein C resistance --> intermediate type PhenotypicFeature --> output type Gene\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>input</th>\n",
       "      <th>input_type</th>\n",
       "      <th>pred1</th>\n",
       "      <th>pred1_source</th>\n",
       "      <th>pred1_api</th>\n",
       "      <th>pred1_pubmed</th>\n",
       "      <th>node1_type</th>\n",
       "      <th>node1_name</th>\n",
       "      <th>node1_id</th>\n",
       "      <th>pred2</th>\n",
       "      <th>pred2_source</th>\n",
       "      <th>pred2_api</th>\n",
       "      <th>pred2_pubmed</th>\n",
       "      <th>output_type</th>\n",
       "      <th>output_name</th>\n",
       "      <th>output_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>treated_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>9974416</td>\n",
       "      <td>Gene</td>\n",
       "      <td>C1333075</td>\n",
       "      <td>UMLS:C1333075</td>\n",
       "      <td>positively_regulates</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Gene API</td>\n",
       "      <td>10048117</td>\n",
       "      <td>Gene</td>\n",
       "      <td>C0128902</td>\n",
       "      <td>UMLS:C0128902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>treated_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>9974416</td>\n",
       "      <td>Gene</td>\n",
       "      <td>C1333075</td>\n",
       "      <td>UMLS:C1333075</td>\n",
       "      <td>physically_interacts_with</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Gene API</td>\n",
       "      <td>27210873</td>\n",
       "      <td>Gene</td>\n",
       "      <td>F9</td>\n",
       "      <td>NCBIGene:2158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>disrupted_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>14515684</td>\n",
       "      <td>Gene</td>\n",
       "      <td>C2985367</td>\n",
       "      <td>UMLS:C2985367</td>\n",
       "      <td>positively_regulates</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Gene API</td>\n",
       "      <td>7541242</td>\n",
       "      <td>Gene</td>\n",
       "      <td>F9</td>\n",
       "      <td>NCBIGene:2158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>affected_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>16879211,17549295,8618815,8701401,8956934</td>\n",
       "      <td>Gene</td>\n",
       "      <td>C2985367</td>\n",
       "      <td>UMLS:C2985367</td>\n",
       "      <td>positively_regulates</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Gene API</td>\n",
       "      <td>7541242</td>\n",
       "      <td>Gene</td>\n",
       "      <td>F9</td>\n",
       "      <td>NCBIGene:2158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>related_to</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>28717431</td>\n",
       "      <td>Gene</td>\n",
       "      <td>C2985367</td>\n",
       "      <td>UMLS:C2985367</td>\n",
       "      <td>positively_regulates</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Gene API</td>\n",
       "      <td>7541242</td>\n",
       "      <td>Gene</td>\n",
       "      <td>F9</td>\n",
       "      <td>NCBIGene:2158</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            input input_type         pred1 pred1_source  \\\n",
       "0  ACTIVATED PROTEIN C RESISTANCE    Disease    treated_by       SEMMED   \n",
       "1  ACTIVATED PROTEIN C RESISTANCE    Disease    treated_by       SEMMED   \n",
       "2  ACTIVATED PROTEIN C RESISTANCE    Disease  disrupted_by       SEMMED   \n",
       "3  ACTIVATED PROTEIN C RESISTANCE    Disease   affected_by       SEMMED   \n",
       "4  ACTIVATED PROTEIN C RESISTANCE    Disease    related_to       SEMMED   \n",
       "\n",
       "            pred1_api                               pred1_pubmed node1_type  \\\n",
       "0  SEMMED Disease API                                    9974416       Gene   \n",
       "1  SEMMED Disease API                                    9974416       Gene   \n",
       "2  SEMMED Disease API                                   14515684       Gene   \n",
       "3  SEMMED Disease API  16879211,17549295,8618815,8701401,8956934       Gene   \n",
       "4  SEMMED Disease API                                   28717431       Gene   \n",
       "\n",
       "  node1_name       node1_id                      pred2 pred2_source  \\\n",
       "0   C1333075  UMLS:C1333075       positively_regulates       SEMMED   \n",
       "1   C1333075  UMLS:C1333075  physically_interacts_with       SEMMED   \n",
       "2   C2985367  UMLS:C2985367       positively_regulates       SEMMED   \n",
       "3   C2985367  UMLS:C2985367       positively_regulates       SEMMED   \n",
       "4   C2985367  UMLS:C2985367       positively_regulates       SEMMED   \n",
       "\n",
       "         pred2_api pred2_pubmed output_type output_name      output_id  \n",
       "0  SEMMED Gene API     10048117        Gene    C0128902  UMLS:C0128902  \n",
       "1  SEMMED Gene API     27210873        Gene          F9  NCBIGene:2158  \n",
       "2  SEMMED Gene API      7541242        Gene          F9  NCBIGene:2158  \n",
       "3  SEMMED Gene API      7541242        Gene          F9  NCBIGene:2158  \n",
       "4  SEMMED Gene API      7541242        Gene          F9  NCBIGene:2158  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get results for disease -> any node type -> gene\n",
    "disease_to_all_nodes_to_genes = predict_many([disease],['Gene'], ALL_NODE_TYPES)\n",
    "disease_to_all_nodes_to_genes.head() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Stored 'disease_to_all_nodes_to_genes' (DataFrame)\n"
     ]
    }
   ],
   "source": [
    "# Save\n",
    "%store disease_to_all_nodes_to_genes\n",
    "%store -r disease_to_all_nodes_to_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>input</th>\n",
       "      <th>input_type</th>\n",
       "      <th>pred1</th>\n",
       "      <th>pred1_source</th>\n",
       "      <th>pred1_api</th>\n",
       "      <th>pred1_pubmed</th>\n",
       "      <th>node1_type</th>\n",
       "      <th>node1_name</th>\n",
       "      <th>node1_id</th>\n",
       "      <th>pred2</th>\n",
       "      <th>pred2_source</th>\n",
       "      <th>pred2_api</th>\n",
       "      <th>pred2_pubmed</th>\n",
       "      <th>output_type</th>\n",
       "      <th>output_name</th>\n",
       "      <th>output_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1663</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>disrupted_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>12624624</td>\n",
       "      <td>Gene</td>\n",
       "      <td>C1705784</td>\n",
       "      <td>UMLS:C1705784</td>\n",
       "      <td>negatively_regulates</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Gene API</td>\n",
       "      <td>27041584</td>\n",
       "      <td>Gene</td>\n",
       "      <td>U2AF2</td>\n",
       "      <td>NCBIGene:11338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1664</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>disrupted_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>12624624</td>\n",
       "      <td>Gene</td>\n",
       "      <td>C1705784</td>\n",
       "      <td>UMLS:C1705784</td>\n",
       "      <td>physically_interacts_with</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Gene API</td>\n",
       "      <td>27041584</td>\n",
       "      <td>Gene</td>\n",
       "      <td>U2AF2</td>\n",
       "      <td>NCBIGene:11338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1665</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>caused_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>9405896</td>\n",
       "      <td>Gene</td>\n",
       "      <td>C0017337</td>\n",
       "      <td>UMLS:C0017337</td>\n",
       "      <td>negatively_regulates</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Gene API</td>\n",
       "      <td>23152763</td>\n",
       "      <td>Gene</td>\n",
       "      <td>U2AF2</td>\n",
       "      <td>NCBIGene:11338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1666</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>disrupted_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>12624624</td>\n",
       "      <td>Gene</td>\n",
       "      <td>CD82</td>\n",
       "      <td>NCBIGene:3732</td>\n",
       "      <td>negatively_regulates</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Gene API</td>\n",
       "      <td>27041584</td>\n",
       "      <td>Gene</td>\n",
       "      <td>U2AF2</td>\n",
       "      <td>NCBIGene:11338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1667</th>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>Disease</td>\n",
       "      <td>disrupted_by</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Disease API</td>\n",
       "      <td>12624624</td>\n",
       "      <td>Gene</td>\n",
       "      <td>CD82</td>\n",
       "      <td>NCBIGene:3732</td>\n",
       "      <td>physically_interacts_with</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Gene API</td>\n",
       "      <td>27041584</td>\n",
       "      <td>Gene</td>\n",
       "      <td>U2AF2</td>\n",
       "      <td>NCBIGene:11338</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               input input_type         pred1 pred1_source  \\\n",
       "1663  ACTIVATED PROTEIN C RESISTANCE    Disease  disrupted_by       SEMMED   \n",
       "1664  ACTIVATED PROTEIN C RESISTANCE    Disease  disrupted_by       SEMMED   \n",
       "1665  ACTIVATED PROTEIN C RESISTANCE    Disease     caused_by       SEMMED   \n",
       "1666  ACTIVATED PROTEIN C RESISTANCE    Disease  disrupted_by       SEMMED   \n",
       "1667  ACTIVATED PROTEIN C RESISTANCE    Disease  disrupted_by       SEMMED   \n",
       "\n",
       "               pred1_api pred1_pubmed node1_type node1_name       node1_id  \\\n",
       "1663  SEMMED Disease API     12624624       Gene   C1705784  UMLS:C1705784   \n",
       "1664  SEMMED Disease API     12624624       Gene   C1705784  UMLS:C1705784   \n",
       "1665  SEMMED Disease API      9405896       Gene   C0017337  UMLS:C0017337   \n",
       "1666  SEMMED Disease API     12624624       Gene       CD82  NCBIGene:3732   \n",
       "1667  SEMMED Disease API     12624624       Gene       CD82  NCBIGene:3732   \n",
       "\n",
       "                          pred2 pred2_source        pred2_api pred2_pubmed  \\\n",
       "1663       negatively_regulates       SEMMED  SEMMED Gene API     27041584   \n",
       "1664  physically_interacts_with       SEMMED  SEMMED Gene API     27041584   \n",
       "1665       negatively_regulates       SEMMED  SEMMED Gene API     23152763   \n",
       "1666       negatively_regulates       SEMMED  SEMMED Gene API     27041584   \n",
       "1667  physically_interacts_with       SEMMED  SEMMED Gene API     27041584   \n",
       "\n",
       "     output_type output_name       output_id  \n",
       "1663        Gene       U2AF2  NCBIGene:11338  \n",
       "1664        Gene       U2AF2  NCBIGene:11338  \n",
       "1665        Gene       U2AF2  NCBIGene:11338  \n",
       "1666        Gene       U2AF2  NCBIGene:11338  \n",
       "1667        Gene       U2AF2  NCBIGene:11338  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# remove entries with symptoms as intermediates\n",
    "indices_with_symptom_intermediates = [i for i, val in enumerate(list(disease_to_all_nodes_to_genes['node1_name'])) if val in disease_symptoms]\n",
    "disease_to_all_nodes_to_genes = disease_to_all_nodes_to_genes.drop( disease_to_all_nodes_to_genes.index[indices_with_symptom_intermediates])\n",
    "# remove UMLS entries - not totally gene specific\n",
    "disease_to_all_nodes_to_genes = disease_to_all_nodes_to_genes[~disease_to_all_nodes_to_genes['output_id'].str.contains('UMLS')]\n",
    "disease_to_all_nodes_to_genes.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Top 10 Gene Occurrences : \n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'TNF': 472,\n",
       " 'VEGFA': 423,\n",
       " 'AKT1': 313,\n",
       " 'ESR1': 311,\n",
       " 'INS': 283,\n",
       " 'TP53': 268,\n",
       " 'MMP9': 267,\n",
       " 'TGFB1': 259,\n",
       " 'EGFR': 255,\n",
       " 'CAT': 255}"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create dictionary of gene reults whereby each Gene has values for the number of occurrences of the gene in results\n",
    "# and the number of publications connecting the disease -> intermediates -> gene\n",
    "disease_to_all_nodes_to_genes_results = {}\n",
    "disease_to_all_nodes_to_genes_genes = list(disease_to_all_nodes_to_genes[\"output_name\"]) # create list of genes\n",
    "disease_to_all_nodes_to_genes_genes = list(dict.fromkeys(disease_to_all_nodes_to_genes_genes))  # remove duplicates\n",
    "\n",
    "for gene in disease_to_all_nodes_to_genes_genes: \n",
    "    disease_to_all_nodes_to_genes_results[gene] = {\n",
    "        'gene_count' : 0,\n",
    "        'publications': []\n",
    "    }\n",
    "\n",
    "for index, row in disease_to_all_nodes_to_genes.iterrows():\n",
    "    disease_to_all_nodes_to_genes_results[row['output_name']]['gene_count'] = disease_to_all_nodes_to_genes_results[row['output_name']]['gene_count'] + 1\n",
    "    if(row['pred1_pubmed'] != None): \n",
    "        disease_to_all_nodes_to_genes_results[row['output_name']]['publications'] = disease_to_all_nodes_to_genes_results[row['output_name']]['publications'] + row['pred1_pubmed'].split(\",\")\n",
    "    if(row['pred2_pubmed'] != None): \n",
    "        disease_to_all_nodes_to_genes_results[row['output_name']]['publications'] = disease_to_all_nodes_to_genes_results[row['output_name']]['publications'] + row['pred2_pubmed'].split(\",\")\n",
    "\n",
    "disease_to_all_nodes_to_genes_results = dict(sorted(disease_to_all_nodes_to_genes_results.items(), key = lambda x: x[1]['gene_count'], reverse = True))\n",
    "# disease_to_all_nodes_to_genes_results\n",
    "\n",
    "for key,value in disease_to_all_nodes_to_genes_results.items(): \n",
    "    disease_to_all_nodes_to_genes_results[key]['publications'] = list(dict.fromkeys(disease_to_all_nodes_to_genes_results[key]['publications']))\n",
    "    \n",
    "        \n",
    "# printing top 10   \n",
    "print(\"Top 10 Gene Occurrences : \")\n",
    "{A:N['gene_count'] for (A,N) in [x for x in disease_to_all_nodes_to_genes_results.items()][:10]}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Determine Genes to Further Analyze "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CDSN',\n",
       " 'TFPI',\n",
       " 'F5',\n",
       " 'SSB',\n",
       " 'PROS1',\n",
       " 'CD82',\n",
       " 'KLK3',\n",
       " 'SERPINE1',\n",
       " 'SERPINB2',\n",
       " 'JAK2',\n",
       " 'TWIST1',\n",
       " 'TUBB4B',\n",
       " 'NEUROD1',\n",
       " 'CP',\n",
       " 'GPI',\n",
       " 'F8',\n",
       " 'PLAT',\n",
       " 'PLG',\n",
       " 'PROC',\n",
       " 'LPA',\n",
       " 'TNF',\n",
       " 'VEGFA',\n",
       " 'AKT1',\n",
       " 'ESR1',\n",
       " 'INS',\n",
       " 'TP53',\n",
       " 'MMP9',\n",
       " 'TGFB1',\n",
       " 'EGFR',\n",
       " 'CAT',\n",
       " 'MMP2',\n",
       " 'LEP',\n",
       " 'IL6',\n",
       " 'CA2',\n",
       " 'CAMP',\n",
       " 'MTHFR',\n",
       " 'CD4',\n",
       " 'AR',\n",
       " 'IFNA1',\n",
       " 'MAPK1',\n",
       " 'FAS',\n",
       " 'PTH',\n",
       " 'STAT3',\n",
       " 'FOS',\n",
       " 'APP',\n",
       " 'CDKN1A',\n",
       " 'ALB',\n",
       " 'IL2',\n",
       " 'VWF',\n",
       " 'MPO',\n",
       " 'TLR4',\n",
       " 'CCL2',\n",
       " 'POMC',\n",
       " 'LPL',\n",
       " 'MAPK8',\n",
       " 'REN',\n",
       " 'EGF',\n",
       " 'LOX',\n",
       " 'CISH',\n",
       " 'CCND1',\n",
       " 'IL10',\n",
       " 'CD40',\n",
       " 'APOB',\n",
       " 'CRK',\n",
       " 'ACE',\n",
       " 'TH',\n",
       " 'ESR2',\n",
       " 'TIMP1',\n",
       " 'BAX',\n",
       " 'PGR',\n",
       " 'WDTC1',\n",
       " 'TCEAL1',\n",
       " 'GRK2',\n",
       " 'HSPA4',\n",
       " 'ATM',\n",
       " 'CD36',\n",
       " 'PTGS2',\n",
       " 'ERVK-10',\n",
       " 'PIK3CA',\n",
       " 'H3P16',\n",
       " 'MTOR',\n",
       " 'RHO',\n",
       " 'ABCB1',\n",
       " 'KIT',\n",
       " 'PTEN',\n",
       " 'PPARA',\n",
       " 'CXCR4',\n",
       " 'CDKN2A',\n",
       " 'IGF1',\n",
       " 'RELA',\n",
       " 'ADAMTS13',\n",
       " 'GABPA',\n",
       " 'CD44',\n",
       " 'TAT',\n",
       " 'ANG',\n",
       " 'PGF',\n",
       " 'TTR',\n",
       " 'MYC',\n",
       " 'ITGB2',\n",
       " 'BRCA1',\n",
       " 'ADA',\n",
       " 'RHOA',\n",
       " 'STAT5A',\n",
       " 'ITGAM',\n",
       " 'TF',\n",
       " 'EGR1',\n",
       " 'SST',\n",
       " 'MAPK3',\n",
       " 'SIRT1']"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get list of genes based off of \"max\" gene parameters\n",
    "disease_top_genes_list = ([A for (A,N) in [x for x in disease_to_gene_results.items()][:max_one_step_genes]]\n",
    "    + [A for (A,N) in [x for x in disease_to_all_nodes_to_genes_results.items()][:max_two_step_genes]])\n",
    "\n",
    "disease_top_genes_list = list(dict.fromkeys(disease_top_genes_list))\n",
    "disease_top_genes_list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. Get Disease Symptoms related to Genes\n",
    "Genes -> Symptoms, then filter based on disease symptoms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "APP Failed\n",
      "[{'NCBIGene': '1041', 'name': 'corneodesmosin', 'SYMBOL': 'CDSN', 'UMLS': 'C1413299', 'HGNC': '1802', 'UNIPROTKB': 'Q15517', 'ENSEMBL': 'ENSG00000137197', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '1041'}, 'display': 'NCBIGene(1041) ENSEMBL(ENSG00000137197) HGNC(1802) UMLS(C1413299) UNIPROTKB(Q15517) SYMBOL(CDSN)', 'type': 'Gene'}, {'NCBIGene': '7035', 'name': 'tissue factor pathway inhibitor', 'SYMBOL': 'TFPI', 'UMLS': 'C1420705', 'HGNC': '11760', 'UNIPROTKB': 'P10646', 'ENSEMBL': 'ENSG00000003436', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '7035'}, 'display': 'NCBIGene(7035) ENSEMBL(ENSG00000003436) HGNC(11760) UMLS(C1420705) UNIPROTKB(P10646) SYMBOL(TFPI)', 'type': 'Gene'}, {'NCBIGene': '2153', 'name': 'coagulation factor V', 'SYMBOL': 'F5', 'UMLS': 'C1414509', 'HGNC': '3542', 'UNIPROTKB': 'P12259', 'ENSEMBL': 'ENSG00000198734', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '2153'}, 'display': 'NCBIGene(2153) ENSEMBL(ENSG00000198734) HGNC(3542) UMLS(C1414509) UNIPROTKB(P12259) SYMBOL(F5)', 'type': 'Gene'}, {'NCBIGene': '6741', 'name': 'small RNA binding exonuclease protection factor La', 'SYMBOL': 'SSB', 'UMLS': 'C1420423', 'HGNC': '11316', 'UNIPROTKB': 'P05455', 'ENSEMBL': 'ENSG00000138385', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '6741'}, 'display': 'NCBIGene(6741) ENSEMBL(ENSG00000138385) HGNC(11316) UMLS(C1420423) UNIPROTKB(P05455) SYMBOL(SSB)', 'type': 'Gene'}, {'NCBIGene': '5627', 'name': 'protein S', 'SYMBOL': 'PROS1', 'UMLS': 'C1418948', 'HGNC': '9456', 'UNIPROTKB': 'P07225', 'ENSEMBL': 'ENSG00000184500', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5627'}, 'display': 'NCBIGene(5627) ENSEMBL(ENSG00000184500) HGNC(9456) UMLS(C1418948) UNIPROTKB(P07225) SYMBOL(PROS1)', 'type': 'Gene'}, {'NCBIGene': '3732', 'name': 'CD82 molecule', 'SYMBOL': 'CD82', 'UMLS': 'C0812444', 'HGNC': '6210', 'UNIPROTKB': 'P27701', 'ENSEMBL': 'ENSG00000085117', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '3732'}, 'display': 'NCBIGene(3732) ENSEMBL(ENSG00000085117) HGNC(6210) UMLS(C0812444) UNIPROTKB(P27701) SYMBOL(CD82)', 'type': 'Gene'}, {'NCBIGene': '354', 'name': 'kallikrein related peptidase 3', 'SYMBOL': 'KLK3', 'UMLS': 'C1366489', 'HGNC': '6364', 'UNIPROTKB': 'P07288', 'ENSEMBL': 'ENSG00000142515', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '354'}, 'display': 'NCBIGene(354) ENSEMBL(ENSG00000142515) HGNC(6364) UMLS(C1366489) UNIPROTKB(P07288) SYMBOL(KLK3)', 'type': 'Gene'}, {'NCBIGene': '5054', 'name': 'serpin family E member 1', 'SYMBOL': 'SERPINE1', 'UMLS': 'C1418252', 'HGNC': '8583', 'UNIPROTKB': 'P05121', 'ENSEMBL': 'ENSG00000106366', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5054'}, 'display': 'NCBIGene(5054) ENSEMBL(ENSG00000106366) HGNC(8583) UMLS(C1418252) UNIPROTKB(P05121) SYMBOL(SERPINE1)', 'type': 'Gene'}, {'NCBIGene': '5055', 'name': 'serpin family B member 2', 'SYMBOL': 'SERPINB2', 'UMLS': 'C1418253', 'HGNC': '8584', 'UNIPROTKB': 'P05120', 'ENSEMBL': 'ENSG00000197632', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5055'}, 'display': 'NCBIGene(5055) ENSEMBL(ENSG00000197632) HGNC(8584) UMLS(C1418253) UNIPROTKB(P05120) SYMBOL(SERPINB2)', 'type': 'Gene'}, {'NCBIGene': '3717', 'name': 'Janus kinase 2', 'SYMBOL': 'JAK2', 'UMLS': 'C1334291', 'HGNC': '6192', 'UNIPROTKB': 'O60674', 'ENSEMBL': 'ENSG00000096968', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '3717'}, 'display': 'NCBIGene(3717) ENSEMBL(ENSG00000096968) HGNC(6192) UMLS(C1334291) UNIPROTKB(O60674) SYMBOL(JAK2)', 'type': 'Gene'}, {'NCBIGene': '7291', 'name': 'twist family bHLH transcription factor 1', 'SYMBOL': 'TWIST1', 'UMLS': 'C1421244', 'HGNC': '12428', 'UNIPROTKB': 'Q15672', 'ENSEMBL': 'ENSG00000122691', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '7291'}, 'display': 'NCBIGene(7291) ENSEMBL(ENSG00000122691) HGNC(12428) UMLS(C1421244) UNIPROTKB(Q15672) SYMBOL(TWIST1)', 'type': 'Gene'}, {'NCBIGene': '10383', 'name': 'tubulin beta 4B class IVb', 'SYMBOL': 'TUBB4B', 'UMLS': 'C1823554', 'HGNC': '20771', 'UNIPROTKB': 'P68371', 'ENSEMBL': 'ENSG00000188229', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '10383'}, 'display': 'NCBIGene(10383) ENSEMBL(ENSG00000188229) HGNC(20771) UMLS(C1823554) UNIPROTKB(P68371) SYMBOL(TUBB4B)', 'type': 'Gene'}, {'NCBIGene': '4760', 'name': 'neuronal differentiation 1', 'SYMBOL': 'NEUROD1', 'UMLS': 'C1417683', 'HGNC': '7762', 'UNIPROTKB': 'Q13562', 'ENSEMBL': 'ENSG00000162992', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '4760'}, 'display': 'NCBIGene(4760) ENSEMBL(ENSG00000162992) HGNC(7762) UMLS(C1417683) UNIPROTKB(Q13562) SYMBOL(NEUROD1)', 'type': 'Gene'}, {'NCBIGene': '1356', 'name': 'ceruloplasmin', 'SYMBOL': 'CP', 'UMLS': 'C1439306', 'HGNC': '2295', 'UNIPROTKB': 'P00450', 'ENSEMBL': 'ENSG00000047457', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '1356'}, 'display': 'NCBIGene(1356) ENSEMBL(ENSG00000047457) HGNC(2295) UMLS(C1439306) UNIPROTKB(P00450) SYMBOL(CP)', 'type': 'Gene'}, {'NCBIGene': '2821', 'name': 'glucose-6-phosphate isomerase', 'SYMBOL': 'GPI', 'UMLS': 'C1333690', 'HGNC': '4458', 'UNIPROTKB': 'P06744', 'ENSEMBL': 'ENSG00000105220', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '2821'}, 'display': 'NCBIGene(2821) ENSEMBL(ENSG00000105220) HGNC(4458) UMLS(C1333690) UNIPROTKB(P06744) SYMBOL(GPI)', 'type': 'Gene'}, {'NCBIGene': '2157', 'name': 'coagulation factor VIII', 'SYMBOL': 'F8', 'UMLS': 'C1366370', 'HGNC': '3546', 'UNIPROTKB': 'P00451', 'ENSEMBL': 'ENSG00000185010', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '2157'}, 'display': 'NCBIGene(2157) ENSEMBL(ENSG00000185010) HGNC(3546) UMLS(C1366370) UNIPROTKB(P00451) SYMBOL(F8)', 'type': 'Gene'}, {'NCBIGene': '5327', 'name': 'plasminogen activator, tissue type', 'SYMBOL': 'PLAT', 'UMLS': 'C1367601', 'HGNC': '9051', 'UNIPROTKB': 'P00750', 'ENSEMBL': 'ENSG00000104368', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5327'}, 'display': 'NCBIGene(5327) ENSEMBL(ENSG00000104368) HGNC(9051) UMLS(C1367601) UNIPROTKB(P00750) SYMBOL(PLAT)', 'type': 'Gene'}, {'NCBIGene': '5340', 'name': 'plasminogen', 'SYMBOL': 'PLG', 'UMLS': 'C1367553', 'HGNC': '9071', 'UNIPROTKB': 'P00747', 'ENSEMBL': 'ENSG00000122194', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5340'}, 'display': 'NCBIGene(5340) ENSEMBL(ENSG00000122194) HGNC(9071) UMLS(C1367553) UNIPROTKB(P00747) SYMBOL(PLG)', 'type': 'Gene'}, {'NCBIGene': '5624', 'name': 'protein C, inactivator of coagulation factors Va and VIIIa', 'SYMBOL': 'PROC', 'UMLS': 'C1418943', 'HGNC': '9451', 'UNIPROTKB': 'P04070', 'ENSEMBL': 'ENSG00000115718', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5624'}, 'display': 'NCBIGene(5624) ENSEMBL(ENSG00000115718) HGNC(9451) UMLS(C1418943) UNIPROTKB(P04070) SYMBOL(PROC)', 'type': 'Gene'}, {'NCBIGene': '4018', 'name': 'lipoprotein(a)', 'SYMBOL': 'LPA', 'UMLS': 'C1439335', 'HGNC': '6667', 'ENSEMBL': 'ENSG00000198670', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '4018'}, 'display': 'NCBIGene(4018) ENSEMBL(ENSG00000198670) HGNC(6667) UMLS(C1439335) SYMBOL(LPA)', 'type': 'Gene'}, {'NCBIGene': '7124', 'name': 'tumor necrosis factor', 'SYMBOL': 'TNF', 'UMLS': 'C0812246', 'HGNC': '11892', 'UNIPROTKB': 'P01375', 'ENSEMBL': 'ENSG00000204490', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '7124'}, 'display': 'NCBIGene(7124) ENSEMBL(ENSG00000204490) HGNC(11892) UMLS(C0812246) UNIPROTKB(P01375) SYMBOL(TNF)', 'type': 'Gene'}, {'NCBIGene': '7422', 'name': 'vascular endothelial growth factor A', 'SYMBOL': 'VEGFA', 'UMLS': 'C1823619', 'HGNC': '12680', 'UNIPROTKB': 'P15692', 'ENSEMBL': 'ENSG00000112715', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '7422'}, 'display': 'NCBIGene(7422) ENSEMBL(ENSG00000112715) HGNC(12680) UMLS(C1823619) UNIPROTKB(P15692) SYMBOL(VEGFA)', 'type': 'Gene'}, {'NCBIGene': '207', 'name': 'AKT serine/threonine kinase 1', 'SYMBOL': 'AKT1', 'UMLS': 'C0812228', 'HGNC': '391', 'UNIPROTKB': 'P31749', 'ENSEMBL': 'ENSG00000142208', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '207'}, 'display': 'NCBIGene(207) ENSEMBL(ENSG00000142208) HGNC(391) UMLS(C0812228) UNIPROTKB(P31749) SYMBOL(AKT1)', 'type': 'Gene'}, {'NCBIGene': '2099', 'name': 'estrogen receptor 1', 'SYMBOL': 'ESR1', 'UMLS': 'C1414461', 'HGNC': '3467', 'UNIPROTKB': 'P03372', 'ENSEMBL': 'ENSG00000091831', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '2099'}, 'display': 'NCBIGene(2099) ENSEMBL(ENSG00000091831) HGNC(3467) UMLS(C1414461) UNIPROTKB(P03372) SYMBOL(ESR1)', 'type': 'Gene'}, {'NCBIGene': '3630', 'name': 'insulin', 'SYMBOL': 'INS', 'UMLS': 'C1337112', 'HGNC': '6081', 'UNIPROTKB': 'P01308', 'ENSEMBL': 'ENSG00000254647', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '3630'}, 'display': 'NCBIGene(3630) ENSEMBL(ENSG00000254647) HGNC(6081) UMLS(C1337112) UNIPROTKB(P01308) SYMBOL(INS)', 'type': 'Gene'}, {'NCBIGene': '7157', 'name': 'tumor protein p53', 'SYMBOL': 'TP53', 'UMLS': 'C0079419', 'HGNC': '11998', 'UNIPROTKB': 'P04637', 'ENSEMBL': 'ENSG00000141510', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '7157'}, 'display': 'NCBIGene(7157) ENSEMBL(ENSG00000141510) HGNC(11998) UMLS(C0079419) UNIPROTKB(P04637) SYMBOL(TP53)', 'type': 'Gene'}, {'NCBIGene': '4318', 'name': 'matrix metallopeptidase 9', 'SYMBOL': 'MMP9', 'UMLS': 'C1334523', 'HGNC': '7176', 'UNIPROTKB': 'P14780', 'ENSEMBL': 'ENSG00000100985', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '4318'}, 'display': 'NCBIGene(4318) ENSEMBL(ENSG00000100985) HGNC(7176) UMLS(C1334523) UNIPROTKB(P14780) SYMBOL(MMP9)', 'type': 'Gene'}, {'NCBIGene': '7040', 'name': 'transforming growth factor beta 1', 'SYMBOL': 'TGFB1', 'UMLS': 'C1366557', 'HGNC': '11766', 'UNIPROTKB': 'P01137', 'ENSEMBL': 'ENSG00000105329', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '7040'}, 'display': 'NCBIGene(7040) ENSEMBL(ENSG00000105329) HGNC(11766) UMLS(C1366557) UNIPROTKB(P01137) SYMBOL(TGFB1)', 'type': 'Gene'}, {'NCBIGene': '1956', 'name': 'epidermal growth factor receptor', 'SYMBOL': 'EGFR', 'UMLS': 'C1414313', 'HGNC': '3236', 'UNIPROTKB': 'P00533', 'ENSEMBL': 'ENSG00000146648', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '1956'}, 'display': 'NCBIGene(1956) ENSEMBL(ENSG00000146648) HGNC(3236) UMLS(C1414313) UNIPROTKB(P00533) SYMBOL(EGFR)', 'type': 'Gene'}, {'NCBIGene': '847', 'name': 'catalase', 'SYMBOL': 'CAT', 'UMLS': 'C1413138', 'HGNC': '1516', 'UNIPROTKB': 'P04040', 'ENSEMBL': 'ENSG00000121691', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '847'}, 'display': 'NCBIGene(847) ENSEMBL(ENSG00000121691) HGNC(1516) UMLS(C1413138) UNIPROTKB(P04040) SYMBOL(CAT)', 'type': 'Gene'}, {'NCBIGene': '4313', 'name': 'matrix metallopeptidase 2', 'SYMBOL': 'MMP2', 'UMLS': 'C1334522', 'HGNC': '7166', 'UNIPROTKB': 'P08253', 'ENSEMBL': 'ENSG00000087245', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '4313'}, 'display': 'NCBIGene(4313) ENSEMBL(ENSG00000087245) HGNC(7166) UMLS(C1334522) UNIPROTKB(P08253) SYMBOL(MMP2)', 'type': 'Gene'}, {'NCBIGene': '3952', 'name': 'leptin', 'SYMBOL': 'LEP', 'UMLS': 'C1416825', 'HGNC': '6553', 'UNIPROTKB': 'P41159', 'ENSEMBL': 'ENSG00000174697', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '3952'}, 'display': 'NCBIGene(3952) ENSEMBL(ENSG00000174697) HGNC(6553) UMLS(C1416825) UNIPROTKB(P41159) SYMBOL(LEP)', 'type': 'Gene'}, {'NCBIGene': '3569', 'name': 'interleukin 6', 'SYMBOL': 'IL6', 'UMLS': 'C1334122', 'HGNC': '6018', 'UNIPROTKB': 'P05231', 'ENSEMBL': 'ENSG00000136244', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '3569'}, 'display': 'NCBIGene(3569) ENSEMBL(ENSG00000136244) HGNC(6018) UMLS(C1334122) UNIPROTKB(P05231) SYMBOL(IL6)', 'type': 'Gene'}, {'NCBIGene': '760', 'name': 'carbonic anhydrase 2', 'SYMBOL': 'CA2', 'UMLS': 'C1413043', 'HGNC': '1373', 'UNIPROTKB': 'P00918', 'ENSEMBL': 'ENSG00000104267', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '760'}, 'display': 'NCBIGene(760) ENSEMBL(ENSG00000104267) HGNC(1373) UMLS(C1413043) UNIPROTKB(P00918) SYMBOL(CA2)', 'type': 'Gene'}, {'NCBIGene': '820', 'name': 'cathelicidin antimicrobial peptide', 'SYMBOL': 'CAMP', 'UMLS': 'C1413106', 'HGNC': '1472', 'UNIPROTKB': 'P49913', 'ENSEMBL': 'ENSG00000164047', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '820'}, 'display': 'NCBIGene(820) ENSEMBL(ENSG00000164047) HGNC(1472) UMLS(C1413106) UNIPROTKB(P49913) SYMBOL(CAMP)', 'type': 'Gene'}, {'NCBIGene': '4524', 'name': 'methylenetetrahydrofolate reductase', 'SYMBOL': 'MTHFR', 'UMLS': 'C0919427', 'HGNC': '7436', 'UNIPROTKB': 'P42898', 'ENSEMBL': 'ENSG00000177000', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '4524'}, 'display': 'NCBIGene(4524) ENSEMBL(ENSG00000177000) HGNC(7436) UMLS(C0919427) UNIPROTKB(P42898) SYMBOL(MTHFR)', 'type': 'Gene'}, {'NCBIGene': '920', 'name': 'CD4 molecule', 'SYMBOL': 'CD4', 'UMLS': 'C1332714', 'HGNC': '1678', 'UNIPROTKB': 'P01730', 'ENSEMBL': 'ENSG00000010610', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '920'}, 'display': 'NCBIGene(920) ENSEMBL(ENSG00000010610) HGNC(1678) UMLS(C1332714) UNIPROTKB(P01730) SYMBOL(CD4)', 'type': 'Gene'}, {'NCBIGene': '367', 'name': 'androgen receptor', 'SYMBOL': 'AR', 'UMLS': 'C1367578', 'HGNC': '644', 'UNIPROTKB': 'P10275', 'ENSEMBL': 'ENSG00000169083', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '367'}, 'display': 'NCBIGene(367) ENSEMBL(ENSG00000169083) HGNC(644) UMLS(C1367578) UNIPROTKB(P10275) SYMBOL(AR)', 'type': 'Gene'}, {'NCBIGene': '3439', 'name': 'interferon alpha 1', 'SYMBOL': 'IFNA1', 'UMLS': 'C1415900', 'HGNC': '5417', 'UNIPROTKB': 'P01562', 'ENSEMBL': 'ENSG00000197919', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '3439'}, 'display': 'NCBIGene(3439) ENSEMBL(ENSG00000197919) HGNC(5417) UMLS(C1415900) UNIPROTKB(P01562) SYMBOL(IFNA1)', 'type': 'Gene'}, {'NCBIGene': '5594', 'name': 'mitogen-activated protein kinase 1', 'SYMBOL': 'MAPK1', 'UMLS': 'C1366882', 'HGNC': '6871', 'UNIPROTKB': 'P28482', 'ENSEMBL': 'ENSG00000100030', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5594'}, 'display': 'NCBIGene(5594) ENSEMBL(ENSG00000100030) HGNC(6871) UMLS(C1366882) UNIPROTKB(P28482) SYMBOL(MAPK1)', 'type': 'Gene'}, {'NCBIGene': '355', 'name': 'Fas cell surface death receptor', 'SYMBOL': 'FAS', 'UMLS': 'C1539477', 'HGNC': '11920', 'UNIPROTKB': 'P25445', 'ENSEMBL': 'ENSG00000026103', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '355'}, 'display': 'NCBIGene(355) ENSEMBL(ENSG00000026103) HGNC(11920) UMLS(C1539477) UNIPROTKB(P25445) SYMBOL(FAS)', 'type': 'Gene'}, {'NCBIGene': '5741', 'name': 'parathyroid hormone', 'SYMBOL': 'PTH', 'UMLS': 'C1419072', 'HGNC': '9606', 'UNIPROTKB': 'P01270', 'ENSEMBL': 'ENSG00000152266', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5741'}, 'display': 'NCBIGene(5741) ENSEMBL(ENSG00000152266) HGNC(9606) UMLS(C1419072) UNIPROTKB(P01270) SYMBOL(PTH)', 'type': 'Gene'}, {'NCBIGene': '6774', 'name': 'signal transducer and activator of transcription 3', 'SYMBOL': 'STAT3', 'UMLS': 'C1367307', 'HGNC': '11364', 'UNIPROTKB': 'P40763', 'ENSEMBL': 'ENSG00000168610', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '6774'}, 'display': 'NCBIGene(6774) ENSEMBL(ENSG00000168610) HGNC(11364) UMLS(C1367307) UNIPROTKB(P40763) SYMBOL(STAT3)', 'type': 'Gene'}, {'NCBIGene': '2353', 'name': 'Fos proto-oncogene, AP-1 transcription factor subunit', 'SYMBOL': 'FOS', 'UMLS': 'C0087140', 'HGNC': '3796', 'UNIPROTKB': 'P01100', 'ENSEMBL': 'ENSG00000170345', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '2353'}, 'display': 'NCBIGene(2353) ENSEMBL(ENSG00000170345) HGNC(3796) UMLS(C0087140) UNIPROTKB(P01100) SYMBOL(FOS)', 'type': 'Gene'}, {'NCBIGene': '1026', 'name': 'cyclin dependent kinase inhibitor 1A', 'SYMBOL': 'CDKN1A', 'UMLS': 'C0249197', 'HGNC': '1784', 'UNIPROTKB': 'P38936', 'ENSEMBL': 'ENSG00000124762', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '1026'}, 'display': 'NCBIGene(1026) ENSEMBL(ENSG00000124762) HGNC(1784) UMLS(C0249197) UNIPROTKB(P38936) SYMBOL(CDKN1A)', 'type': 'Gene'}, {'NCBIGene': '213', 'name': 'albumin', 'SYMBOL': 'ALB', 'UMLS': 'C1412332', 'HGNC': '399', 'UNIPROTKB': 'P02768', 'ENSEMBL': 'ENSG00000163631', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '213'}, 'display': 'NCBIGene(213) ENSEMBL(ENSG00000163631) HGNC(399) UMLS(C1412332) UNIPROTKB(P02768) SYMBOL(ALB)', 'type': 'Gene'}, {'NCBIGene': '3558', 'name': 'interleukin 2', 'SYMBOL': 'IL2', 'UMLS': 'C0879590', 'HGNC': '6001', 'UNIPROTKB': 'P60568', 'ENSEMBL': 'ENSG00000109471', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '3558'}, 'display': 'NCBIGene(3558) ENSEMBL(ENSG00000109471) HGNC(6001) UMLS(C0879590) UNIPROTKB(P60568) SYMBOL(IL2)', 'type': 'Gene'}, {'NCBIGene': '7450', 'name': 'von Willebrand factor', 'SYMBOL': 'VWF', 'UMLS': 'C1336938', 'HGNC': '12726', 'UNIPROTKB': 'P04275', 'ENSEMBL': 'ENSG00000110799', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '7450'}, 'display': 'NCBIGene(7450) ENSEMBL(ENSG00000110799) HGNC(12726) UMLS(C1336938) UNIPROTKB(P04275) SYMBOL(VWF)', 'type': 'Gene'}, {'NCBIGene': '4353', 'name': 'myeloperoxidase', 'SYMBOL': 'MPO', 'UMLS': 'C1417245', 'HGNC': '7218', 'UNIPROTKB': 'P05164', 'ENSEMBL': 'ENSG00000005381', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '4353'}, 'display': 'NCBIGene(4353) ENSEMBL(ENSG00000005381) HGNC(7218) UMLS(C1417245) UNIPROTKB(P05164) SYMBOL(MPO)', 'type': 'Gene'}, {'NCBIGene': '7099', 'name': 'toll like receptor 4', 'SYMBOL': 'TLR4', 'UMLS': 'C1336636', 'HGNC': '11850', 'UNIPROTKB': 'O00206', 'ENSEMBL': 'ENSG00000136869', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '7099'}, 'display': 'NCBIGene(7099) ENSEMBL(ENSG00000136869) HGNC(11850) UMLS(C1336636) UNIPROTKB(O00206) SYMBOL(TLR4)', 'type': 'Gene'}, {'NCBIGene': '6347', 'name': 'C-C motif chemokine ligand 2', 'SYMBOL': 'CCL2', 'UMLS': 'C1337092', 'HGNC': '10618', 'UNIPROTKB': 'P13500', 'ENSEMBL': 'ENSG00000108691', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '6347'}, 'display': 'NCBIGene(6347) ENSEMBL(ENSG00000108691) HGNC(10618) UMLS(C1337092) UNIPROTKB(P13500) SYMBOL(CCL2)', 'type': 'Gene'}, {'NCBIGene': '5443', 'name': 'proopiomelanocortin', 'SYMBOL': 'POMC', 'UMLS': 'C1337111', 'HGNC': '9201', 'UNIPROTKB': 'P01189', 'ENSEMBL': 'ENSG00000115138', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5443'}, 'display': 'NCBIGene(5443) ENSEMBL(ENSG00000115138) HGNC(9201) UMLS(C1337111) UNIPROTKB(P01189) SYMBOL(POMC)', 'type': 'Gene'}, {'NCBIGene': '4023', 'name': 'lipoprotein lipase', 'SYMBOL': 'LPL', 'UMLS': 'C1416902', 'HGNC': '6677', 'UNIPROTKB': 'P06858', 'ENSEMBL': 'ENSG00000175445', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '4023'}, 'display': 'NCBIGene(4023) ENSEMBL(ENSG00000175445) HGNC(6677) UMLS(C1416902) UNIPROTKB(P06858) SYMBOL(LPL)', 'type': 'Gene'}, {'NCBIGene': '5599', 'name': 'mitogen-activated protein kinase 8', 'SYMBOL': 'MAPK8', 'UMLS': 'C1367731', 'HGNC': '6881', 'UNIPROTKB': 'P45983', 'ENSEMBL': 'ENSG00000107643', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5599'}, 'display': 'NCBIGene(5599) ENSEMBL(ENSG00000107643) HGNC(6881) UMLS(C1367731) UNIPROTKB(P45983) SYMBOL(MAPK8)', 'type': 'Gene'}, {'NCBIGene': '5972', 'name': 'renin', 'SYMBOL': 'REN', 'UMLS': 'C1419338', 'HGNC': '9958', 'UNIPROTKB': 'P00797', 'ENSEMBL': 'ENSG00000143839', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5972'}, 'display': 'NCBIGene(5972) ENSEMBL(ENSG00000143839) HGNC(9958) UMLS(C1419338) UNIPROTKB(P00797) SYMBOL(REN)', 'type': 'Gene'}, {'NCBIGene': '1950', 'name': 'epidermal growth factor', 'SYMBOL': 'EGF', 'UMLS': 'C1367721', 'HGNC': '3229', 'UNIPROTKB': 'P01133', 'ENSEMBL': 'ENSG00000138798', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '1950'}, 'display': 'NCBIGene(1950) ENSEMBL(ENSG00000138798) HGNC(3229) UMLS(C1367721) UNIPROTKB(P01133) SYMBOL(EGF)', 'type': 'Gene'}, {'NCBIGene': '4015', 'name': 'lysyl oxidase', 'SYMBOL': 'LOX', 'UMLS': 'C1334349', 'HGNC': '6664', 'UNIPROTKB': 'P28300', 'ENSEMBL': 'ENSG00000113083', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '4015'}, 'display': 'NCBIGene(4015) ENSEMBL(ENSG00000113083) HGNC(6664) UMLS(C1334349) UNIPROTKB(P28300) SYMBOL(LOX)', 'type': 'Gene'}, {'NCBIGene': '1154', 'name': 'cytokine inducible SH2 containing protein', 'SYMBOL': 'CISH', 'UMLS': 'C1332762', 'HGNC': '1984', 'UNIPROTKB': 'Q9NSE2', 'ENSEMBL': 'ENSG00000114737', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '1154'}, 'display': 'NCBIGene(1154) ENSEMBL(ENSG00000114737) HGNC(1984) UMLS(C1332762) UNIPROTKB(Q9NSE2) SYMBOL(CISH)', 'type': 'Gene'}, {'NCBIGene': '595', 'name': 'cyclin D1', 'SYMBOL': 'CCND1', 'UMLS': 'C1413172', 'HGNC': '1582', 'UNIPROTKB': 'P24385', 'ENSEMBL': 'ENSG00000110092', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '595'}, 'display': 'NCBIGene(595) ENSEMBL(ENSG00000110092) HGNC(1582) UMLS(C1413172) UNIPROTKB(P24385) SYMBOL(CCND1)', 'type': 'Gene'}, {'NCBIGene': '3586', 'name': 'interleukin 10', 'SYMBOL': 'IL10', 'UMLS': 'C1334098', 'HGNC': '5962', 'UNIPROTKB': 'P22301', 'ENSEMBL': 'ENSG00000136634', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '3586'}, 'display': 'NCBIGene(3586) ENSEMBL(ENSG00000136634) HGNC(5962) UMLS(C1334098) UNIPROTKB(P22301) SYMBOL(IL10)', 'type': 'Gene'}, {'NCBIGene': '958', 'name': 'CD40 molecule', 'SYMBOL': 'CD40', 'UMLS': 'C1539081', 'HGNC': '11919', 'UNIPROTKB': 'P25942', 'ENSEMBL': 'ENSG00000101017', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '958'}, 'display': 'NCBIGene(958) ENSEMBL(ENSG00000101017) HGNC(11919) UMLS(C1539081) UNIPROTKB(P25942) SYMBOL(CD40)', 'type': 'Gene'}, {'NCBIGene': '338', 'name': 'apolipoprotein B', 'SYMBOL': 'APOB', 'UMLS': 'C1412471', 'HGNC': '603', 'UNIPROTKB': 'P04114', 'ENSEMBL': 'ENSG00000084674', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '338'}, 'display': 'NCBIGene(338) ENSEMBL(ENSG00000084674) HGNC(603) UMLS(C1412471) UNIPROTKB(P04114) SYMBOL(APOB)', 'type': 'Gene'}, {'NCBIGene': '1398', 'name': 'CRK proto-oncogene, adaptor protein', 'SYMBOL': 'CRK', 'UMLS': 'C0812252', 'HGNC': '2362', 'UNIPROTKB': 'P46108', 'ENSEMBL': 'ENSG00000167193', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '1398'}, 'display': 'NCBIGene(1398) ENSEMBL(ENSG00000167193) HGNC(2362) UMLS(C0812252) UNIPROTKB(P46108) SYMBOL(CRK)', 'type': 'Gene'}, {'NCBIGene': '1636', 'name': 'angiotensin I converting enzyme', 'SYMBOL': 'ACE', 'UMLS': 'C1413931', 'HGNC': '2707', 'UNIPROTKB': 'P12821', 'ENSEMBL': 'ENSG00000159640', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '1636'}, 'display': 'NCBIGene(1636) ENSEMBL(ENSG00000159640) HGNC(2707) UMLS(C1413931) UNIPROTKB(P12821) SYMBOL(ACE)', 'type': 'Gene'}, {'NCBIGene': '7054', 'name': 'tyrosine hydroxylase', 'SYMBOL': 'TH', 'UMLS': 'C1420718', 'HGNC': '11782', 'UNIPROTKB': 'P07101', 'ENSEMBL': 'ENSG00000180176', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '7054'}, 'display': 'NCBIGene(7054) ENSEMBL(ENSG00000180176) HGNC(11782) UMLS(C1420718) UNIPROTKB(P07101) SYMBOL(TH)', 'type': 'Gene'}, {'NCBIGene': '2100', 'name': 'estrogen receptor 2', 'SYMBOL': 'ESR2', 'UMLS': 'C1414462', 'HGNC': '3468', 'UNIPROTKB': 'Q92731', 'ENSEMBL': 'ENSG00000140009', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '2100'}, 'display': 'NCBIGene(2100) ENSEMBL(ENSG00000140009) HGNC(3468) UMLS(C1414462) UNIPROTKB(Q92731) SYMBOL(ESR2)', 'type': 'Gene'}, {'NCBIGene': '7076', 'name': 'TIMP metallopeptidase inhibitor 1', 'SYMBOL': 'TIMP1', 'UMLS': 'C1367459', 'HGNC': '11820', 'UNIPROTKB': 'P01033', 'ENSEMBL': 'ENSG00000102265', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '7076'}, 'display': 'NCBIGene(7076) ENSEMBL(ENSG00000102265) HGNC(11820) UMLS(C1367459) UNIPROTKB(P01033) SYMBOL(TIMP1)', 'type': 'Gene'}, {'NCBIGene': '581', 'name': 'BCL2 associated X, apoptosis regulator', 'SYMBOL': 'BAX', 'UMLS': 'C0812198', 'HGNC': '959', 'UNIPROTKB': 'Q07812', 'ENSEMBL': 'ENSG00000087088', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '581'}, 'display': 'NCBIGene(581) ENSEMBL(ENSG00000087088) HGNC(959) UMLS(C0812198) UNIPROTKB(Q07812) SYMBOL(BAX)', 'type': 'Gene'}, {'NCBIGene': '5241', 'name': 'progesterone receptor', 'SYMBOL': 'PGR', 'UMLS': 'C1366500', 'HGNC': '8910', 'UNIPROTKB': 'P06401', 'ENSEMBL': 'ENSG00000082175', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5241'}, 'display': 'NCBIGene(5241) ENSEMBL(ENSG00000082175) HGNC(8910) UMLS(C1366500) UNIPROTKB(P06401) SYMBOL(PGR)', 'type': 'Gene'}, {'NCBIGene': '23038', 'name': 'WD and tetratricopeptide repeats 1', 'SYMBOL': 'WDTC1', 'UMLS': 'C1428806', 'HGNC': '29175', 'UNIPROTKB': 'Q8N5D0', 'ENSEMBL': 'ENSG00000142784', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '23038'}, 'display': 'NCBIGene(23038) ENSEMBL(ENSG00000142784) HGNC(29175) UMLS(C1428806) UNIPROTKB(Q8N5D0) SYMBOL(WDTC1)', 'type': 'Gene'}, {'NCBIGene': '9338', 'name': 'transcription elongation factor A like 1', 'SYMBOL': 'TCEAL1', 'UMLS': 'C1420626', 'HGNC': '11616', 'UNIPROTKB': 'Q15170', 'ENSEMBL': 'ENSG00000172465', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '9338'}, 'display': 'NCBIGene(9338) ENSEMBL(ENSG00000172465) HGNC(11616) UMLS(C1420626) UNIPROTKB(Q15170) SYMBOL(TCEAL1)', 'type': 'Gene'}, {'NCBIGene': '156', 'name': 'G protein-coupled receptor kinase 2', 'SYMBOL': 'GRK2', 'UMLS': 'C1366490', 'HGNC': '289', 'UNIPROTKB': 'P25098', 'ENSEMBL': 'ENSG00000173020', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '156'}, 'display': 'NCBIGene(156) ENSEMBL(ENSG00000173020) HGNC(289) UMLS(C1366490) UNIPROTKB(P25098) SYMBOL(GRK2)', 'type': 'Gene'}, {'NCBIGene': '3308', 'name': 'heat shock protein family A (Hsp70) member 4', 'SYMBOL': 'HSPA4', 'UMLS': 'C1415760', 'HGNC': '5237', 'UNIPROTKB': 'P34932', 'ENSEMBL': 'ENSG00000170606', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '3308'}, 'display': 'NCBIGene(3308) ENSEMBL(ENSG00000170606) HGNC(5237) UMLS(C1415760) UNIPROTKB(P34932) SYMBOL(HSPA4)', 'type': 'Gene'}, {'NCBIGene': '472', 'name': 'ATM serine/threonine kinase', 'SYMBOL': 'ATM', 'UMLS': 'C0919524', 'HGNC': '795', 'UNIPROTKB': 'Q13315', 'ENSEMBL': 'ENSG00000149311', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '472'}, 'display': 'NCBIGene(472) ENSEMBL(ENSG00000149311) HGNC(795) UMLS(C0919524) UNIPROTKB(Q13315) SYMBOL(ATM)', 'type': 'Gene'}, {'NCBIGene': '948', 'name': 'CD36 molecule', 'SYMBOL': 'CD36', 'UMLS': 'C1366645', 'HGNC': '1663', 'UNIPROTKB': 'P16671', 'ENSEMBL': 'ENSG00000135218', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '948'}, 'display': 'NCBIGene(948) ENSEMBL(ENSG00000135218) HGNC(1663) UMLS(C1366645) UNIPROTKB(P16671) SYMBOL(CD36)', 'type': 'Gene'}, {'NCBIGene': '5743', 'name': 'prostaglandin-endoperoxide synthase 2', 'SYMBOL': 'PTGS2', 'UMLS': 'C1367485', 'HGNC': '9605', 'UNIPROTKB': 'P35354', 'ENSEMBL': 'ENSG00000073756', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5743'}, 'display': 'NCBIGene(5743) ENSEMBL(ENSG00000073756) HGNC(9605) UMLS(C1367485) UNIPROTKB(P35354) SYMBOL(PTGS2)', 'type': 'Gene'}, {'NCBIGene': '100616101', 'name': 'endogenous retrovirus group K member 10', 'SYMBOL': 'ERVK-10', 'UMLS': 'C3147205', 'HGNC': '39004', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '100616101'}, 'display': 'NCBIGene(100616101) HGNC(39004) UMLS(C3147205) SYMBOL(ERVK-10)', 'type': 'Gene'}, {'NCBIGene': '5290', 'name': 'phosphatidylinositol-4,5-bisphosphate 3-kinase catalytic subunit alpha', 'SYMBOL': 'PIK3CA', 'UMLS': 'C1335212', 'HGNC': '8975', 'UNIPROTKB': 'P42336', 'ENSEMBL': 'ENSG00000121879', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5290'}, 'display': 'NCBIGene(5290) ENSEMBL(ENSG00000121879) HGNC(8975) UMLS(C1335212) UNIPROTKB(P42336) SYMBOL(PIK3CA)', 'type': 'Gene'}, {'name': 'H3 histone pseudogene 16', 'SYMBOL': 'H3P16', 'ENSEMBL': 'ENSG00000178458', 'primary': {'identifier': 'ENSEMBL', 'cls': 'Gene', 'value': 'ENSG00000178458'}, 'display': 'ENSEMBL(ENSG00000178458) SYMBOL(H3P16)', 'type': 'Gene'}, {'NCBIGene': '2475', 'name': 'mechanistic target of rapamycin kinase', 'SYMBOL': 'MTOR', 'UMLS': 'C1414805', 'HGNC': '3942', 'UNIPROTKB': 'P42345', 'ENSEMBL': 'ENSG00000198793', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '2475'}, 'display': 'NCBIGene(2475) ENSEMBL(ENSG00000198793) HGNC(3942) UMLS(C1414805) UNIPROTKB(P42345) SYMBOL(MTOR)', 'type': 'Gene'}, {'NCBIGene': '6010', 'name': 'rhodopsin', 'SYMBOL': 'RHO', 'UMLS': 'C1419385', 'HGNC': '10012', 'UNIPROTKB': 'P08100', 'ENSEMBL': 'ENSG00000163914', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '6010'}, 'display': 'NCBIGene(6010) ENSEMBL(ENSG00000163914) HGNC(10012) UMLS(C1419385) UNIPROTKB(P08100) SYMBOL(RHO)', 'type': 'Gene'}, {'NCBIGene': '5243', 'name': 'ATP binding cassette subfamily B member 1', 'SYMBOL': 'ABCB1', 'UMLS': 'C0376622', 'HGNC': '40', 'UNIPROTKB': 'P08183', 'ENSEMBL': 'ENSG00000085563', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5243'}, 'display': 'NCBIGene(5243) ENSEMBL(ENSG00000085563) HGNC(40) UMLS(C0376622) UNIPROTKB(P08183) SYMBOL(ABCB1)', 'type': 'Gene'}, {'NCBIGene': '3815', 'name': 'KIT proto-oncogene, receptor tyrosine kinase', 'SYMBOL': 'KIT', 'UMLS': 'C0920288', 'HGNC': '6342', 'UNIPROTKB': 'P10721', 'ENSEMBL': 'ENSG00000157404', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '3815'}, 'display': 'NCBIGene(3815) ENSEMBL(ENSG00000157404) HGNC(6342) UMLS(C0920288) UNIPROTKB(P10721) SYMBOL(KIT)', 'type': 'Gene'}, {'NCBIGene': '5728', 'name': 'phosphatase and tensin homolog', 'SYMBOL': 'PTEN', 'UMLS': 'C0694888', 'HGNC': '9588', 'UNIPROTKB': 'P60484', 'ENSEMBL': 'ENSG00000171862', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5728'}, 'display': 'NCBIGene(5728) ENSEMBL(ENSG00000171862) HGNC(9588) UMLS(C0694888) UNIPROTKB(P60484) SYMBOL(PTEN)', 'type': 'Gene'}, {'NCBIGene': '5465', 'name': 'peroxisome proliferator activated receptor alpha', 'SYMBOL': 'PPARA', 'UMLS': 'C1418776', 'HGNC': '9232', 'UNIPROTKB': 'Q07869', 'ENSEMBL': 'ENSG00000186951', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5465'}, 'display': 'NCBIGene(5465) ENSEMBL(ENSG00000186951) HGNC(9232) UMLS(C1418776) UNIPROTKB(Q07869) SYMBOL(PPARA)', 'type': 'Gene'}, {'NCBIGene': '7852', 'name': 'C-X-C motif chemokine receptor 4', 'SYMBOL': 'CXCR4', 'UMLS': 'C1332823', 'HGNC': '2561', 'UNIPROTKB': 'P61073', 'ENSEMBL': 'ENSG00000121966', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '7852'}, 'display': 'NCBIGene(7852) ENSEMBL(ENSG00000121966) HGNC(2561) UMLS(C1332823) UNIPROTKB(P61073) SYMBOL(CXCR4)', 'type': 'Gene'}, {'NCBIGene': '1029', 'name': 'cyclin dependent kinase inhibitor 2A', 'SYMBOL': 'CDKN2A', 'UMLS': 'C0525037', 'HGNC': '1787', 'UNIPROTKB': 'P42771', 'ENSEMBL': 'ENSG00000147889', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '1029'}, 'display': 'NCBIGene(1029) ENSEMBL(ENSG00000147889) HGNC(1787) UMLS(C0525037) UNIPROTKB(P42771) SYMBOL(CDKN2A)', 'type': 'Gene'}, {'NCBIGene': '3479', 'name': 'insulin like growth factor 1', 'SYMBOL': 'IGF1', 'UMLS': 'C1334089', 'HGNC': '5464', 'UNIPROTKB': 'P05019', 'ENSEMBL': 'ENSG00000017427', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '3479'}, 'display': 'NCBIGene(3479) ENSEMBL(ENSG00000017427) HGNC(5464) UMLS(C1334089) UNIPROTKB(P05019) SYMBOL(IGF1)', 'type': 'Gene'}, {'NCBIGene': '5970', 'name': 'RELA proto-oncogene, NF-kB subunit', 'SYMBOL': 'RELA', 'UMLS': 'C1335610', 'HGNC': '9955', 'UNIPROTKB': 'Q04206', 'ENSEMBL': 'ENSG00000173039', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5970'}, 'display': 'NCBIGene(5970) ENSEMBL(ENSG00000173039) HGNC(9955) UMLS(C1335610) UNIPROTKB(Q04206) SYMBOL(RELA)', 'type': 'Gene'}, {'NCBIGene': '11093', 'name': 'ADAM metallopeptidase with thrombospondin type 1 motif 13', 'SYMBOL': 'ADAMTS13', 'UMLS': 'C1413036', 'HGNC': '1366', 'UNIPROTKB': 'Q76LX8', 'ENSEMBL': 'ENSG00000160323', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '11093'}, 'display': 'NCBIGene(11093) ENSEMBL(ENSG00000160323) HGNC(1366) UMLS(C1413036) UNIPROTKB(Q76LX8) SYMBOL(ADAMTS13)', 'type': 'Gene'}, {'NCBIGene': '2551', 'name': 'GA binding protein transcription factor subunit alpha', 'SYMBOL': 'GABPA', 'UMLS': 'C1414905', 'HGNC': '4071', 'UNIPROTKB': 'Q06546', 'ENSEMBL': 'ENSG00000154727', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '2551'}, 'display': 'NCBIGene(2551) ENSEMBL(ENSG00000154727) HGNC(4071) UMLS(C1414905) UNIPROTKB(Q06546) SYMBOL(GABPA)', 'type': 'Gene'}, {'NCBIGene': '960', 'name': 'CD44 molecule (Indian blood group)', 'SYMBOL': 'CD44', 'UMLS': 'C1332712', 'HGNC': '1681', 'UNIPROTKB': 'P16070', 'ENSEMBL': 'ENSG00000026508', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '960'}, 'display': 'NCBIGene(960) ENSEMBL(ENSG00000026508) HGNC(1681) UMLS(C1332712) UNIPROTKB(P16070) SYMBOL(CD44)', 'type': 'Gene'}, {'NCBIGene': '6898', 'name': 'tyrosine aminotransferase', 'SYMBOL': 'TAT', 'UMLS': 'C1420590', 'HGNC': '11573', 'UNIPROTKB': 'P17735', 'ENSEMBL': 'ENSG00000198650', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '6898'}, 'display': 'NCBIGene(6898) ENSEMBL(ENSG00000198650) HGNC(11573) UMLS(C1420590) UNIPROTKB(P17735) SYMBOL(TAT)', 'type': 'Gene'}, {'NCBIGene': '283', 'name': 'angiogenin', 'SYMBOL': 'ANG', 'UMLS': 'C1367482', 'HGNC': '483', 'UNIPROTKB': 'P03950', 'ENSEMBL': 'ENSG00000214274', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '283'}, 'display': 'NCBIGene(283) ENSEMBL(ENSG00000214274) HGNC(483) UMLS(C1367482) UNIPROTKB(P03950) SYMBOL(ANG)', 'type': 'Gene'}, {'NCBIGene': '5228', 'name': 'placental growth factor', 'SYMBOL': 'PGF', 'UMLS': 'C1335207', 'HGNC': '8893', 'UNIPROTKB': 'P49763', 'ENSEMBL': 'ENSG00000119630', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5228'}, 'display': 'NCBIGene(5228) ENSEMBL(ENSG00000119630) HGNC(8893) UMLS(C1335207) UNIPROTKB(P49763) SYMBOL(PGF)', 'type': 'Gene'}, {'NCBIGene': '7276', 'name': 'transthyretin', 'SYMBOL': 'TTR', 'UMLS': 'C1421224', 'HGNC': '12405', 'UNIPROTKB': 'P02766', 'ENSEMBL': 'ENSG00000118271', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '7276'}, 'display': 'NCBIGene(7276) ENSEMBL(ENSG00000118271) HGNC(12405) UMLS(C1421224) UNIPROTKB(P02766) SYMBOL(TTR)', 'type': 'Gene'}, {'NCBIGene': '4609', 'name': 'MYC proto-oncogene, bHLH transcription factor', 'SYMBOL': 'MYC', 'UMLS': 'C0086661', 'HGNC': '7553', 'UNIPROTKB': 'P01106', 'ENSEMBL': 'ENSG00000136997', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '4609'}, 'display': 'NCBIGene(4609) ENSEMBL(ENSG00000136997) HGNC(7553) UMLS(C0086661) UNIPROTKB(P01106) SYMBOL(MYC)', 'type': 'Gene'}, {'NCBIGene': '3689', 'name': 'integrin subunit beta 2', 'SYMBOL': 'ITGB2', 'UMLS': 'C1334146', 'HGNC': '6155', 'UNIPROTKB': 'P05107', 'ENSEMBL': 'ENSG00000160255', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '3689'}, 'display': 'NCBIGene(3689) ENSEMBL(ENSG00000160255) HGNC(6155) UMLS(C1334146) UNIPROTKB(P05107) SYMBOL(ITGB2)', 'type': 'Gene'}, {'NCBIGene': '672', 'name': 'BRCA1 DNA repair associated', 'SYMBOL': 'BRCA1', 'UMLS': 'C0376571', 'HGNC': '1100', 'UNIPROTKB': 'P38398', 'ENSEMBL': 'ENSG00000012048', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '672'}, 'display': 'NCBIGene(672) ENSEMBL(ENSG00000012048) HGNC(1100) UMLS(C0376571) UNIPROTKB(P38398) SYMBOL(BRCA1)', 'type': 'Gene'}, {'NCBIGene': '100', 'name': 'adenosine deaminase', 'SYMBOL': 'ADA', 'UMLS': 'C1412179', 'HGNC': '186', 'UNIPROTKB': 'P00813', 'ENSEMBL': 'ENSG00000196839', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '100'}, 'display': 'NCBIGene(100) ENSEMBL(ENSG00000196839) HGNC(186) UMLS(C1412179) UNIPROTKB(P00813) SYMBOL(ADA)', 'type': 'Gene'}, {'NCBIGene': '387', 'name': 'ras homolog family member A', 'SYMBOL': 'RHOA', 'UMLS': 'C0812234', 'HGNC': '667', 'UNIPROTKB': 'P61586', 'ENSEMBL': 'ENSG00000067560', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '387'}, 'display': 'NCBIGene(387) ENSEMBL(ENSG00000067560) HGNC(667) UMLS(C0812234) UNIPROTKB(P61586) SYMBOL(RHOA)', 'type': 'Gene'}, {'NCBIGene': '6776', 'name': 'signal transducer and activator of transcription 5A', 'SYMBOL': 'STAT5A', 'UMLS': 'C1335875', 'HGNC': '11366', 'UNIPROTKB': 'P42229', 'ENSEMBL': 'ENSG00000126561', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '6776'}, 'display': 'NCBIGene(6776) ENSEMBL(ENSG00000126561) HGNC(11366) UMLS(C1335875) UNIPROTKB(P42229) SYMBOL(STAT5A)', 'type': 'Gene'}, {'NCBIGene': '3684', 'name': 'integrin subunit alpha M', 'SYMBOL': 'ITGAM', 'UMLS': 'C1366562', 'HGNC': '6149', 'UNIPROTKB': 'P11215', 'ENSEMBL': 'ENSG00000169896', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '3684'}, 'display': 'NCBIGene(3684) ENSEMBL(ENSG00000169896) HGNC(6149) UMLS(C1366562) UNIPROTKB(P11215) SYMBOL(ITGAM)', 'type': 'Gene'}, {'NCBIGene': '7018', 'name': 'transferrin', 'SYMBOL': 'TF', 'UMLS': 'C1442762', 'HGNC': '11740', 'UNIPROTKB': 'P02787', 'ENSEMBL': 'ENSG00000091513', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '7018'}, 'display': 'NCBIGene(7018) ENSEMBL(ENSG00000091513) HGNC(11740) UMLS(C1442762) UNIPROTKB(P02787) SYMBOL(TF)', 'type': 'Gene'}, {'NCBIGene': '1958', 'name': 'early growth response 1', 'SYMBOL': 'EGR1', 'UMLS': 'C1414314', 'HGNC': '3238', 'UNIPROTKB': 'P18146', 'ENSEMBL': 'ENSG00000120738', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '1958'}, 'display': 'NCBIGene(1958) ENSEMBL(ENSG00000120738) HGNC(3238) UMLS(C1414314) UNIPROTKB(P18146) SYMBOL(EGR1)', 'type': 'Gene'}, {'NCBIGene': '6750', 'name': 'somatostatin', 'SYMBOL': 'SST', 'UMLS': 'C1335871', 'HGNC': '11329', 'UNIPROTKB': 'P61278', 'ENSEMBL': 'ENSG00000157005', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '6750'}, 'display': 'NCBIGene(6750) ENSEMBL(ENSG00000157005) HGNC(11329) UMLS(C1335871) UNIPROTKB(P61278) SYMBOL(SST)', 'type': 'Gene'}, {'NCBIGene': '5595', 'name': 'mitogen-activated protein kinase 3', 'SYMBOL': 'MAPK3', 'UMLS': 'C1366765', 'HGNC': '6877', 'UNIPROTKB': 'P27361', 'ENSEMBL': 'ENSG00000102882', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5595'}, 'display': 'NCBIGene(5595) ENSEMBL(ENSG00000102882) HGNC(6877) UMLS(C1366765) UNIPROTKB(P27361) SYMBOL(MAPK3)', 'type': 'Gene'}, {'NCBIGene': '23411', 'name': 'sirtuin 1', 'SYMBOL': 'SIRT1', 'UMLS': 'C1423062', 'HGNC': '14929', 'UNIPROTKB': 'Q96EB6', 'ENSEMBL': 'ENSG00000096717', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '23411'}, 'display': 'NCBIGene(23411) ENSEMBL(ENSG00000096717) HGNC(14929) UMLS(C1423062) UNIPROTKB(Q96EB6) SYMBOL(SIRT1)', 'type': 'Gene'}]\n"
     ]
    }
   ],
   "source": [
    "# get gene inputs through hint module\n",
    "gene_inputs = []\n",
    "for gene in disease_top_genes_list: \n",
    "    try: \n",
    "        gene_input = ht.query(gene)[\"Gene\"][0]\n",
    "        gene_inputs.append(gene_input)\n",
    "    except: \n",
    "        print(gene + ' Failed')\n",
    "\n",
    "print(gene_inputs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running: corneodesmosin --> output type PhenotypicFeature\n",
      "Running: corneodesmosin --> output type BiologicalProcess\n",
      "Running: corneodesmosin --> output type Disease\n",
      "Running: tissue factor pathway inhibitor --> output type PhenotypicFeature\n",
      "Running: tissue factor pathway inhibitor --> output type BiologicalProcess\n",
      "Running: tissue factor pathway inhibitor --> output type Disease\n",
      "Running: coagulation factor V --> output type PhenotypicFeature\n",
      "Running: coagulation factor V --> output type BiologicalProcess\n",
      "Running: coagulation factor V --> output type Disease\n",
      "Running: small RNA binding exonuclease protection factor La --> output type PhenotypicFeature\n",
      "Running: small RNA binding exonuclease protection factor La --> output type BiologicalProcess\n",
      "Running: small RNA binding exonuclease protection factor La --> output type Disease\n",
      "Running: protein S --> output type PhenotypicFeature\n",
      "Running: protein S --> output type BiologicalProcess\n",
      "Running: protein S --> output type Disease\n",
      "Running: CD82 molecule --> output type PhenotypicFeature\n",
      "Running: CD82 molecule --> output type BiologicalProcess\n",
      "Running: CD82 molecule --> output type Disease\n",
      "Running: kallikrein related peptidase 3 --> output type PhenotypicFeature\n",
      "Running: kallikrein related peptidase 3 --> output type BiologicalProcess\n",
      "Running: kallikrein related peptidase 3 --> output type Disease\n",
      "Running: serpin family E member 1 --> output type PhenotypicFeature\n",
      "Running: serpin family E member 1 --> output type BiologicalProcess\n",
      "Running: serpin family E member 1 --> output type Disease\n",
      "Running: serpin family B member 2 --> output type PhenotypicFeature\n",
      "Running: serpin family B member 2 --> output type BiologicalProcess\n",
      "Running: serpin family B member 2 --> output type Disease\n",
      "Running: Janus kinase 2 --> output type PhenotypicFeature\n",
      "Running: Janus kinase 2 --> output type BiologicalProcess\n",
      "Running: Janus kinase 2 --> output type Disease\n",
      "Running: twist family bHLH transcription factor 1 --> output type PhenotypicFeature\n",
      "Running: twist family bHLH transcription factor 1 --> output type BiologicalProcess\n",
      "Running: twist family bHLH transcription factor 1 --> output type Disease\n",
      "Running: tubulin beta 4B class IVb --> output type PhenotypicFeature\n",
      "Running: tubulin beta 4B class IVb --> output type BiologicalProcess\n",
      "Running: tubulin beta 4B class IVb --> output type Disease\n",
      "Running: neuronal differentiation 1 --> output type PhenotypicFeature\n",
      "Running: neuronal differentiation 1 --> output type BiologicalProcess\n",
      "Running: neuronal differentiation 1 --> output type Disease\n",
      "Running: ceruloplasmin --> output type PhenotypicFeature\n",
      "Running: ceruloplasmin --> output type BiologicalProcess\n",
      "Running: ceruloplasmin --> output type Disease\n",
      "Running: glucose-6-phosphate isomerase --> output type PhenotypicFeature\n",
      "Running: glucose-6-phosphate isomerase --> output type BiologicalProcess\n",
      "Running: glucose-6-phosphate isomerase --> output type Disease\n",
      "Running: coagulation factor VIII --> output type PhenotypicFeature\n",
      "Running: coagulation factor VIII --> output type BiologicalProcess\n",
      "Running: coagulation factor VIII --> output type Disease\n",
      "Running: plasminogen activator, tissue type --> output type PhenotypicFeature\n",
      "Running: plasminogen activator, tissue type --> output type BiologicalProcess\n",
      "Running: plasminogen activator, tissue type --> output type Disease\n",
      "Running: plasminogen --> output type PhenotypicFeature\n",
      "Running: plasminogen --> output type BiologicalProcess\n",
      "Running: plasminogen --> output type Disease\n",
      "Running: protein C, inactivator of coagulation factors Va and VIIIa --> output type PhenotypicFeature\n",
      "Running: protein C, inactivator of coagulation factors Va and VIIIa --> output type BiologicalProcess\n",
      "Running: protein C, inactivator of coagulation factors Va and VIIIa --> output type Disease\n",
      "Running: lipoprotein(a) --> output type PhenotypicFeature\n",
      "Running: lipoprotein(a) --> output type BiologicalProcess\n",
      "Running: lipoprotein(a) --> output type Disease\n",
      "Running: tumor necrosis factor --> output type PhenotypicFeature\n",
      "Running: tumor necrosis factor --> output type BiologicalProcess\n",
      "Running: tumor necrosis factor --> output type Disease\n",
      "Running: vascular endothelial growth factor A --> output type PhenotypicFeature\n",
      "Running: vascular endothelial growth factor A --> output type BiologicalProcess\n",
      "Running: vascular endothelial growth factor A --> output type Disease\n",
      "Running: AKT serine/threonine kinase 1 --> output type PhenotypicFeature\n",
      "Running: AKT serine/threonine kinase 1 --> output type BiologicalProcess\n",
      "Running: AKT serine/threonine kinase 1 --> output type Disease\n",
      "Running: estrogen receptor 1 --> output type PhenotypicFeature\n",
      "Running: estrogen receptor 1 --> output type BiologicalProcess\n",
      "Running: estrogen receptor 1 --> output type Disease\n",
      "Running: insulin --> output type PhenotypicFeature\n",
      "Running: insulin --> output type BiologicalProcess\n",
      "Running: insulin --> output type Disease\n",
      "Running: tumor protein p53 --> output type PhenotypicFeature\n",
      "Running: tumor protein p53 --> output type BiologicalProcess\n",
      "Running: tumor protein p53 --> output type Disease\n",
      "Running: matrix metallopeptidase 9 --> output type PhenotypicFeature\n",
      "Running: matrix metallopeptidase 9 --> output type BiologicalProcess\n",
      "Running: matrix metallopeptidase 9 --> output type Disease\n",
      "Running: transforming growth factor beta 1 --> output type PhenotypicFeature\n",
      "Running: transforming growth factor beta 1 --> output type BiologicalProcess\n",
      "Running: transforming growth factor beta 1 --> output type Disease\n",
      "Running: epidermal growth factor receptor --> output type PhenotypicFeature\n",
      "Running: epidermal growth factor receptor --> output type BiologicalProcess\n",
      "Running: epidermal growth factor receptor --> output type Disease\n",
      "Running: catalase --> output type PhenotypicFeature\n",
      "Running: catalase --> output type BiologicalProcess\n",
      "Running: catalase --> output type Disease\n",
      "Running: matrix metallopeptidase 2 --> output type PhenotypicFeature\n",
      "Running: matrix metallopeptidase 2 --> output type BiologicalProcess\n",
      "Running: matrix metallopeptidase 2 --> output type Disease\n",
      "Running: leptin --> output type PhenotypicFeature\n",
      "Running: leptin --> output type BiologicalProcess\n",
      "Running: leptin --> output type Disease\n",
      "Running: interleukin 6 --> output type PhenotypicFeature\n",
      "Running: interleukin 6 --> output type BiologicalProcess\n",
      "Running: interleukin 6 --> output type Disease\n",
      "Running: carbonic anhydrase 2 --> output type PhenotypicFeature\n",
      "Running: carbonic anhydrase 2 --> output type BiologicalProcess\n",
      "Running: carbonic anhydrase 2 --> output type Disease\n",
      "Running: cathelicidin antimicrobial peptide --> output type PhenotypicFeature\n",
      "Running: cathelicidin antimicrobial peptide --> output type BiologicalProcess\n",
      "Running: cathelicidin antimicrobial peptide --> output type Disease\n",
      "Running: methylenetetrahydrofolate reductase --> output type PhenotypicFeature\n",
      "Running: methylenetetrahydrofolate reductase --> output type BiologicalProcess\n",
      "Running: methylenetetrahydrofolate reductase --> output type Disease\n",
      "Running: CD4 molecule --> output type PhenotypicFeature\n",
      "Running: CD4 molecule --> output type BiologicalProcess\n",
      "Running: CD4 molecule --> output type Disease\n",
      "Running: androgen receptor --> output type PhenotypicFeature\n",
      "Running: androgen receptor --> output type BiologicalProcess\n",
      "Running: androgen receptor --> output type Disease\n",
      "Running: interferon alpha 1 --> output type PhenotypicFeature\n",
      "Running: interferon alpha 1 --> output type BiologicalProcess\n",
      "Running: interferon alpha 1 --> output type Disease\n",
      "Running: mitogen-activated protein kinase 1 --> output type PhenotypicFeature\n",
      "Running: mitogen-activated protein kinase 1 --> output type BiologicalProcess\n",
      "Running: mitogen-activated protein kinase 1 --> output type Disease\n",
      "Running: Fas cell surface death receptor --> output type PhenotypicFeature\n",
      "Running: Fas cell surface death receptor --> output type BiologicalProcess\n",
      "Running: Fas cell surface death receptor --> output type Disease\n",
      "Running: parathyroid hormone --> output type PhenotypicFeature\n",
      "Running: parathyroid hormone --> output type BiologicalProcess\n",
      "Running: parathyroid hormone --> output type Disease\n",
      "Running: signal transducer and activator of transcription 3 --> output type PhenotypicFeature\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running: signal transducer and activator of transcription 3 --> output type BiologicalProcess\n",
      "Running: signal transducer and activator of transcription 3 --> output type Disease\n",
      "Running: Fos proto-oncogene, AP-1 transcription factor subunit --> output type PhenotypicFeature\n",
      "Running: Fos proto-oncogene, AP-1 transcription factor subunit --> output type BiologicalProcess\n",
      "Running: Fos proto-oncogene, AP-1 transcription factor subunit --> output type Disease\n",
      "Running: cyclin dependent kinase inhibitor 1A --> output type PhenotypicFeature\n",
      "Running: cyclin dependent kinase inhibitor 1A --> output type BiologicalProcess\n",
      "Running: cyclin dependent kinase inhibitor 1A --> output type Disease\n",
      "Running: albumin --> output type PhenotypicFeature\n",
      "Running: albumin --> output type BiologicalProcess\n",
      "Running: albumin --> output type Disease\n",
      "Running: interleukin 2 --> output type PhenotypicFeature\n",
      "Running: interleukin 2 --> output type BiologicalProcess\n",
      "Running: interleukin 2 --> output type Disease\n",
      "Running: von Willebrand factor --> output type PhenotypicFeature\n",
      "Running: von Willebrand factor --> output type BiologicalProcess\n",
      "Running: von Willebrand factor --> output type Disease\n",
      "Running: myeloperoxidase --> output type PhenotypicFeature\n",
      "Running: myeloperoxidase --> output type BiologicalProcess\n",
      "Running: myeloperoxidase --> output type Disease\n",
      "Running: toll like receptor 4 --> output type PhenotypicFeature\n",
      "Running: toll like receptor 4 --> output type BiologicalProcess\n",
      "Running: toll like receptor 4 --> output type Disease\n",
      "Running: C-C motif chemokine ligand 2 --> output type PhenotypicFeature\n",
      "Running: C-C motif chemokine ligand 2 --> output type BiologicalProcess\n",
      "Running: C-C motif chemokine ligand 2 --> output type Disease\n",
      "Running: proopiomelanocortin --> output type PhenotypicFeature\n",
      "Running: proopiomelanocortin --> output type BiologicalProcess\n",
      "Running: proopiomelanocortin --> output type Disease\n",
      "Running: lipoprotein lipase --> output type PhenotypicFeature\n",
      "Running: lipoprotein lipase --> output type BiologicalProcess\n",
      "Running: lipoprotein lipase --> output type Disease\n",
      "Running: mitogen-activated protein kinase 8 --> output type PhenotypicFeature\n",
      "Running: mitogen-activated protein kinase 8 --> output type BiologicalProcess\n",
      "Running: mitogen-activated protein kinase 8 --> output type Disease\n",
      "Running: renin --> output type PhenotypicFeature\n",
      "Running: renin --> output type BiologicalProcess\n",
      "Running: renin --> output type Disease\n",
      "Running: epidermal growth factor --> output type PhenotypicFeature\n",
      "Running: epidermal growth factor --> output type BiologicalProcess\n",
      "Running: epidermal growth factor --> output type Disease\n",
      "Running: lysyl oxidase --> output type PhenotypicFeature\n",
      "Running: lysyl oxidase --> output type BiologicalProcess\n",
      "Running: lysyl oxidase --> output type Disease\n",
      "Running: cytokine inducible SH2 containing protein --> output type PhenotypicFeature\n",
      "Running: cytokine inducible SH2 containing protein --> output type BiologicalProcess\n",
      "Running: cytokine inducible SH2 containing protein --> output type Disease\n",
      "Running: cyclin D1 --> output type PhenotypicFeature\n",
      "Running: cyclin D1 --> output type BiologicalProcess\n",
      "Running: cyclin D1 --> output type Disease\n",
      "Running: interleukin 10 --> output type PhenotypicFeature\n",
      "Running: interleukin 10 --> output type BiologicalProcess\n",
      "Running: interleukin 10 --> output type Disease\n",
      "Running: CD40 molecule --> output type PhenotypicFeature\n",
      "Running: CD40 molecule --> output type BiologicalProcess\n",
      "Running: CD40 molecule --> output type Disease\n",
      "Running: apolipoprotein B --> output type PhenotypicFeature\n",
      "Running: apolipoprotein B --> output type BiologicalProcess\n",
      "Running: apolipoprotein B --> output type Disease\n",
      "Running: CRK proto-oncogene, adaptor protein --> output type PhenotypicFeature\n",
      "Running: CRK proto-oncogene, adaptor protein --> output type BiologicalProcess\n",
      "Running: CRK proto-oncogene, adaptor protein --> output type Disease\n",
      "Running: angiotensin I converting enzyme --> output type PhenotypicFeature\n",
      "Running: angiotensin I converting enzyme --> output type BiologicalProcess\n",
      "Running: angiotensin I converting enzyme --> output type Disease\n",
      "Running: tyrosine hydroxylase --> output type PhenotypicFeature\n",
      "Running: tyrosine hydroxylase --> output type BiologicalProcess\n",
      "Running: tyrosine hydroxylase --> output type Disease\n",
      "Running: estrogen receptor 2 --> output type PhenotypicFeature\n",
      "Running: estrogen receptor 2 --> output type BiologicalProcess\n",
      "Running: estrogen receptor 2 --> output type Disease\n",
      "Running: TIMP metallopeptidase inhibitor 1 --> output type PhenotypicFeature\n",
      "Running: TIMP metallopeptidase inhibitor 1 --> output type BiologicalProcess\n",
      "Running: TIMP metallopeptidase inhibitor 1 --> output type Disease\n",
      "Running: BCL2 associated X, apoptosis regulator --> output type PhenotypicFeature\n",
      "Running: BCL2 associated X, apoptosis regulator --> output type BiologicalProcess\n",
      "Running: BCL2 associated X, apoptosis regulator --> output type Disease\n",
      "Running: progesterone receptor --> output type PhenotypicFeature\n",
      "Running: progesterone receptor --> output type BiologicalProcess\n",
      "Running: progesterone receptor --> output type Disease\n",
      "Running: WD and tetratricopeptide repeats 1 --> output type PhenotypicFeature\n",
      "Running: WD and tetratricopeptide repeats 1 --> output type BiologicalProcess\n",
      "Running: WD and tetratricopeptide repeats 1 --> output type Disease\n",
      "Running: transcription elongation factor A like 1 --> output type PhenotypicFeature\n",
      "Running: transcription elongation factor A like 1 --> output type BiologicalProcess\n",
      "Running: transcription elongation factor A like 1 --> output type Disease\n",
      "Running: G protein-coupled receptor kinase 2 --> output type PhenotypicFeature\n",
      "Running: G protein-coupled receptor kinase 2 --> output type BiologicalProcess\n",
      "Running: G protein-coupled receptor kinase 2 --> output type Disease\n",
      "Running: heat shock protein family A (Hsp70) member 4 --> output type PhenotypicFeature\n",
      "Running: heat shock protein family A (Hsp70) member 4 --> output type BiologicalProcess\n",
      "Running: heat shock protein family A (Hsp70) member 4 --> output type Disease\n",
      "Running: ATM serine/threonine kinase --> output type PhenotypicFeature\n",
      "Running: ATM serine/threonine kinase --> output type BiologicalProcess\n",
      "Running: ATM serine/threonine kinase --> output type Disease\n",
      "Running: CD36 molecule --> output type PhenotypicFeature\n",
      "Running: CD36 molecule --> output type BiologicalProcess\n",
      "Running: CD36 molecule --> output type Disease\n",
      "Running: prostaglandin-endoperoxide synthase 2 --> output type PhenotypicFeature\n",
      "Running: prostaglandin-endoperoxide synthase 2 --> output type BiologicalProcess\n",
      "Running: prostaglandin-endoperoxide synthase 2 --> output type Disease\n",
      "Running: endogenous retrovirus group K member 10 --> output type PhenotypicFeature\n",
      "Running: endogenous retrovirus group K member 10 --> output type BiologicalProcess\n",
      "Running: endogenous retrovirus group K member 10 --> output type Disease\n",
      "Running: phosphatidylinositol-4,5-bisphosphate 3-kinase catalytic subunit alpha --> output type PhenotypicFeature\n",
      "Running: phosphatidylinositol-4,5-bisphosphate 3-kinase catalytic subunit alpha --> output type BiologicalProcess\n",
      "Running: phosphatidylinositol-4,5-bisphosphate 3-kinase catalytic subunit alpha --> output type Disease\n",
      "Running: H3 histone pseudogene 16 --> output type PhenotypicFeature\n",
      "Running: H3 histone pseudogene 16 --> output type BiologicalProcess\n",
      "Running: H3 histone pseudogene 16 --> output type Disease\n",
      "Running: mechanistic target of rapamycin kinase --> output type PhenotypicFeature\n",
      "Running: mechanistic target of rapamycin kinase --> output type BiologicalProcess\n",
      "Running: mechanistic target of rapamycin kinase --> output type Disease\n",
      "Running: rhodopsin --> output type PhenotypicFeature\n",
      "Running: rhodopsin --> output type BiologicalProcess\n",
      "Running: rhodopsin --> output type Disease\n",
      "Running: ATP binding cassette subfamily B member 1 --> output type PhenotypicFeature\n",
      "Running: ATP binding cassette subfamily B member 1 --> output type BiologicalProcess\n",
      "Running: ATP binding cassette subfamily B member 1 --> output type Disease\n",
      "Running: KIT proto-oncogene, receptor tyrosine kinase --> output type PhenotypicFeature\n",
      "Running: KIT proto-oncogene, receptor tyrosine kinase --> output type BiologicalProcess\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running: KIT proto-oncogene, receptor tyrosine kinase --> output type Disease\n",
      "Running: phosphatase and tensin homolog --> output type PhenotypicFeature\n",
      "Running: phosphatase and tensin homolog --> output type BiologicalProcess\n",
      "Running: phosphatase and tensin homolog --> output type Disease\n",
      "Running: peroxisome proliferator activated receptor alpha --> output type PhenotypicFeature\n",
      "Running: peroxisome proliferator activated receptor alpha --> output type BiologicalProcess\n",
      "Running: peroxisome proliferator activated receptor alpha --> output type Disease\n",
      "Running: C-X-C motif chemokine receptor 4 --> output type PhenotypicFeature\n",
      "Running: C-X-C motif chemokine receptor 4 --> output type BiologicalProcess\n",
      "Running: C-X-C motif chemokine receptor 4 --> output type Disease\n",
      "Running: cyclin dependent kinase inhibitor 2A --> output type PhenotypicFeature\n",
      "Running: cyclin dependent kinase inhibitor 2A --> output type BiologicalProcess\n",
      "Running: cyclin dependent kinase inhibitor 2A --> output type Disease\n",
      "Running: insulin like growth factor 1 --> output type PhenotypicFeature\n",
      "Running: insulin like growth factor 1 --> output type BiologicalProcess\n",
      "Running: insulin like growth factor 1 --> output type Disease\n",
      "Running: RELA proto-oncogene, NF-kB subunit --> output type PhenotypicFeature\n",
      "Running: RELA proto-oncogene, NF-kB subunit --> output type BiologicalProcess\n",
      "Running: RELA proto-oncogene, NF-kB subunit --> output type Disease\n",
      "Running: ADAM metallopeptidase with thrombospondin type 1 motif 13 --> output type PhenotypicFeature\n",
      "Running: ADAM metallopeptidase with thrombospondin type 1 motif 13 --> output type BiologicalProcess\n",
      "Running: ADAM metallopeptidase with thrombospondin type 1 motif 13 --> output type Disease\n",
      "Running: GA binding protein transcription factor subunit alpha --> output type PhenotypicFeature\n",
      "Running: GA binding protein transcription factor subunit alpha --> output type BiologicalProcess\n",
      "Running: GA binding protein transcription factor subunit alpha --> output type Disease\n",
      "Running: CD44 molecule (Indian blood group) --> output type PhenotypicFeature\n",
      "Running: CD44 molecule (Indian blood group) --> output type BiologicalProcess\n",
      "Running: CD44 molecule (Indian blood group) --> output type Disease\n",
      "Running: tyrosine aminotransferase --> output type PhenotypicFeature\n",
      "Running: tyrosine aminotransferase --> output type BiologicalProcess\n",
      "Running: tyrosine aminotransferase --> output type Disease\n",
      "Running: angiogenin --> output type PhenotypicFeature\n",
      "Running: angiogenin --> output type BiologicalProcess\n",
      "Running: angiogenin --> output type Disease\n",
      "Running: placental growth factor --> output type PhenotypicFeature\n",
      "Running: placental growth factor --> output type BiologicalProcess\n",
      "Running: placental growth factor --> output type Disease\n",
      "Running: transthyretin --> output type PhenotypicFeature\n",
      "Running: transthyretin --> output type BiologicalProcess\n",
      "Running: transthyretin --> output type Disease\n",
      "Running: MYC proto-oncogene, bHLH transcription factor --> output type PhenotypicFeature\n",
      "Running: MYC proto-oncogene, bHLH transcription factor --> output type BiologicalProcess\n",
      "Running: MYC proto-oncogene, bHLH transcription factor --> output type Disease\n",
      "Running: integrin subunit beta 2 --> output type PhenotypicFeature\n",
      "Running: integrin subunit beta 2 --> output type BiologicalProcess\n",
      "Running: integrin subunit beta 2 --> output type Disease\n",
      "Running: BRCA1 DNA repair associated --> output type PhenotypicFeature\n",
      "Running: BRCA1 DNA repair associated --> output type BiologicalProcess\n",
      "Running: BRCA1 DNA repair associated --> output type Disease\n",
      "Running: adenosine deaminase --> output type PhenotypicFeature\n",
      "Running: adenosine deaminase --> output type BiologicalProcess\n",
      "Running: adenosine deaminase --> output type Disease\n",
      "Running: ras homolog family member A --> output type PhenotypicFeature\n",
      "Running: ras homolog family member A --> output type BiologicalProcess\n",
      "Running: ras homolog family member A --> output type Disease\n",
      "Running: signal transducer and activator of transcription 5A --> output type PhenotypicFeature\n",
      "Running: signal transducer and activator of transcription 5A --> output type BiologicalProcess\n",
      "Running: signal transducer and activator of transcription 5A --> output type Disease\n",
      "Running: integrin subunit alpha M --> output type PhenotypicFeature\n",
      "Running: integrin subunit alpha M --> output type BiologicalProcess\n",
      "Running: integrin subunit alpha M --> output type Disease\n",
      "Running: transferrin --> output type PhenotypicFeature\n",
      "Running: transferrin --> output type BiologicalProcess\n",
      "Running: transferrin --> output type Disease\n",
      "Running: early growth response 1 --> output type PhenotypicFeature\n",
      "Running: early growth response 1 --> output type BiologicalProcess\n",
      "Running: early growth response 1 --> output type Disease\n",
      "Running: somatostatin --> output type PhenotypicFeature\n",
      "Running: somatostatin --> output type BiologicalProcess\n",
      "Running: somatostatin --> output type Disease\n",
      "Running: mitogen-activated protein kinase 3 --> output type PhenotypicFeature\n",
      "Running: mitogen-activated protein kinase 3 --> output type BiologicalProcess\n",
      "Running: mitogen-activated protein kinase 3 --> output type Disease\n",
      "Running: sirtuin 1 --> output type PhenotypicFeature\n",
      "Running: sirtuin 1 --> output type BiologicalProcess\n",
      "Running: sirtuin 1 --> output type Disease\n",
      "(156106, 9)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>input</th>\n",
       "      <th>input_type</th>\n",
       "      <th>pred1</th>\n",
       "      <th>pred1_source</th>\n",
       "      <th>pred1_api</th>\n",
       "      <th>pred1_pubmed</th>\n",
       "      <th>output_type</th>\n",
       "      <th>output_name</th>\n",
       "      <th>output_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CDSN</td>\n",
       "      <td>Gene</td>\n",
       "      <td>related_to</td>\n",
       "      <td>None</td>\n",
       "      <td>BioLink API</td>\n",
       "      <td>None</td>\n",
       "      <td>PhenotypicFeature</td>\n",
       "      <td>DECREASED BODY HEIGHT</td>\n",
       "      <td>UMLS:C0349588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CDSN</td>\n",
       "      <td>Gene</td>\n",
       "      <td>related_to</td>\n",
       "      <td>None</td>\n",
       "      <td>BioLink API</td>\n",
       "      <td>None</td>\n",
       "      <td>PhenotypicFeature</td>\n",
       "      <td>BRITTLE HAIR</td>\n",
       "      <td>UMLS:C4072837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CDSN</td>\n",
       "      <td>Gene</td>\n",
       "      <td>related_to</td>\n",
       "      <td>None</td>\n",
       "      <td>BioLink API</td>\n",
       "      <td>None</td>\n",
       "      <td>PhenotypicFeature</td>\n",
       "      <td>DECREASED NUMBER OF SCALP FOLLICLES</td>\n",
       "      <td>UMLS:C1857042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CDSN</td>\n",
       "      <td>Gene</td>\n",
       "      <td>related_to</td>\n",
       "      <td>None</td>\n",
       "      <td>BioLink API</td>\n",
       "      <td>None</td>\n",
       "      <td>PhenotypicFeature</td>\n",
       "      <td>ASTHMA</td>\n",
       "      <td>UMLS:C0004096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CDSN</td>\n",
       "      <td>Gene</td>\n",
       "      <td>related_to</td>\n",
       "      <td>None</td>\n",
       "      <td>BioLink API</td>\n",
       "      <td>None</td>\n",
       "      <td>PhenotypicFeature</td>\n",
       "      <td>ERYTHEMA</td>\n",
       "      <td>UMLS:C0041834</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  input input_type       pred1 pred1_source    pred1_api pred1_pubmed  \\\n",
       "0  CDSN       Gene  related_to         None  BioLink API         None   \n",
       "1  CDSN       Gene  related_to         None  BioLink API         None   \n",
       "2  CDSN       Gene  related_to         None  BioLink API         None   \n",
       "3  CDSN       Gene  related_to         None  BioLink API         None   \n",
       "4  CDSN       Gene  related_to         None  BioLink API         None   \n",
       "\n",
       "         output_type                          output_name      output_id  \n",
       "0  PhenotypicFeature                DECREASED BODY HEIGHT  UMLS:C0349588  \n",
       "1  PhenotypicFeature                         BRITTLE HAIR  UMLS:C4072837  \n",
       "2  PhenotypicFeature  DECREASED NUMBER OF SCALP FOLLICLES  UMLS:C1857042  \n",
       "3  PhenotypicFeature                               ASTHMA  UMLS:C0004096  \n",
       "4  PhenotypicFeature                             ERYTHEMA  UMLS:C0041834  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get genes to symptoms, which could be represented as a phenotypic feature, biological process, or diesase\n",
    "genes_to_symptoms = predict_many(gene_inputs, ['PhenotypicFeature','BiologicalProcess','Disease'])\n",
    "print(genes_to_symptoms.shape)\n",
    "genes_to_symptoms.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>input</th>\n",
       "      <th>input_type</th>\n",
       "      <th>pred1</th>\n",
       "      <th>pred1_source</th>\n",
       "      <th>pred1_api</th>\n",
       "      <th>pred1_pubmed</th>\n",
       "      <th>output_type</th>\n",
       "      <th>output_name</th>\n",
       "      <th>output_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>CDSN</td>\n",
       "      <td>Gene</td>\n",
       "      <td>disrupts</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Gene API</td>\n",
       "      <td>7745572</td>\n",
       "      <td>Disease</td>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>MONDO:MONDO:0008560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>CDSN</td>\n",
       "      <td>Gene</td>\n",
       "      <td>causes</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Gene API</td>\n",
       "      <td>19036061,19764414,19803829</td>\n",
       "      <td>Disease</td>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>MONDO:MONDO:0008560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>CDSN</td>\n",
       "      <td>Gene</td>\n",
       "      <td>related_to</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Gene API</td>\n",
       "      <td>8834124</td>\n",
       "      <td>Disease</td>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>MONDO:MONDO:0008560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>CDSN</td>\n",
       "      <td>Gene</td>\n",
       "      <td>affects</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Gene API</td>\n",
       "      <td>11703344,16493484</td>\n",
       "      <td>Disease</td>\n",
       "      <td>ACTIVATED PROTEIN C RESISTANCE</td>\n",
       "      <td>MONDO:MONDO:0008560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>CDSN</td>\n",
       "      <td>Gene</td>\n",
       "      <td>causes</td>\n",
       "      <td>SEMMED</td>\n",
       "      <td>SEMMED Gene API</td>\n",
       "      <td>2146503</td>\n",
       "      <td>Disease</td>\n",
       "      <td>DEEP VEIN THROMBOSIS</td>\n",
       "      <td>UMLS:C0149871</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    input input_type       pred1 pred1_source        pred1_api  \\\n",
       "119  CDSN       Gene    disrupts       SEMMED  SEMMED Gene API   \n",
       "120  CDSN       Gene      causes       SEMMED  SEMMED Gene API   \n",
       "121  CDSN       Gene  related_to       SEMMED  SEMMED Gene API   \n",
       "122  CDSN       Gene     affects       SEMMED  SEMMED Gene API   \n",
       "151  CDSN       Gene      causes       SEMMED  SEMMED Gene API   \n",
       "\n",
       "                   pred1_pubmed output_type                     output_name  \\\n",
       "119                     7745572     Disease  ACTIVATED PROTEIN C RESISTANCE   \n",
       "120  19036061,19764414,19803829     Disease  ACTIVATED PROTEIN C RESISTANCE   \n",
       "121                     8834124     Disease  ACTIVATED PROTEIN C RESISTANCE   \n",
       "122           11703344,16493484     Disease  ACTIVATED PROTEIN C RESISTANCE   \n",
       "151                     2146503     Disease            DEEP VEIN THROMBOSIS   \n",
       "\n",
       "               output_id  \n",
       "119  MONDO:MONDO:0008560  \n",
       "120  MONDO:MONDO:0008560  \n",
       "121  MONDO:MONDO:0008560  \n",
       "122  MONDO:MONDO:0008560  \n",
       "151        UMLS:C0149871  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# filter gene -> symptoms table to only include rows where symptoms match disease symptoms\n",
    "indices_with_symptom_outputs = [i for i, val in enumerate([x.upper() for x in list(genes_to_symptoms['output_name'])]) if val in disease_symptoms]\n",
    "relevant_genes_to_symptoms_df = genes_to_symptoms.iloc[indices_with_symptom_outputs]\n",
    "relevant_genes_to_symptoms_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Stored 'relevant_genes_to_symptoms_df' (DataFrame)\n"
     ]
    }
   ],
   "source": [
    "%store relevant_genes_to_symptoms_df\n",
    "%store -r relevant_genes_to_symptoms_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create dictionary to keep track of symptoms a gene is related to, and the number of publications relating the \n",
    "# gene to any of the disease symptoms\n",
    "symptoms_results = {}\n",
    "relevant_top_genes_list = list(dict.fromkeys(list(relevant_genes_to_symptoms_df[\"input\"])))\n",
    "for x in relevant_top_genes_list:\n",
    "    symptoms_results[x] = {\n",
    "        \"related_symptoms\" : [],\n",
    "        \"publications\": []\n",
    "    }\n",
    "for index, row in relevant_genes_to_symptoms_df.iterrows():\n",
    "    symptoms_results[row[\"input\"]][\"related_symptoms\"].append(row[\"output_name\"])\n",
    "    if(row[\"pred1_pubmed\"]):\n",
    "        symptoms_results[row[\"input\"]][\"publications\"] = symptoms_results[row[\"input\"]][\"publications\"] + row[\"pred1_pubmed\"].split(',')\n",
    "\n",
    "for key,value in symptoms_results.items(): \n",
    "    symptoms_results[key]['publications'] = list(dict.fromkeys(symptoms_results[key]['publications']))\n",
    "            \n",
    "        \n",
    "# print(symptoms_results)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. Get Genes Edges Out Count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running: corneodesmosin --> output type Gene\n",
      "Running: corneodesmosin --> output type SequenceVariant\n",
      "Running: corneodesmosin --> output type ChemicalSubstance\n",
      "Running: corneodesmosin --> output type Disease\n",
      "Running: corneodesmosin --> output type MolecularActivity\n",
      "Running: corneodesmosin --> output type BiologicalProcess\n",
      "Running: corneodesmosin --> output type CellularComponent\n",
      "Running: corneodesmosin --> output type Pathway\n",
      "Running: corneodesmosin --> output type AnatomicalEntity\n",
      "Running: corneodesmosin --> output type PhenotypicFeature\n",
      "Running: tissue factor pathway inhibitor --> output type Gene\n",
      "Running: tissue factor pathway inhibitor --> output type SequenceVariant\n",
      "Running: tissue factor pathway inhibitor --> output type ChemicalSubstance\n",
      "Running: tissue factor pathway inhibitor --> output type Disease\n",
      "Running: tissue factor pathway inhibitor --> output type MolecularActivity\n",
      "Running: tissue factor pathway inhibitor --> output type BiologicalProcess\n",
      "Running: tissue factor pathway inhibitor --> output type CellularComponent\n",
      "Running: tissue factor pathway inhibitor --> output type Pathway\n",
      "Running: tissue factor pathway inhibitor --> output type AnatomicalEntity\n",
      "Running: tissue factor pathway inhibitor --> output type PhenotypicFeature\n",
      "Running: coagulation factor V --> output type Gene\n",
      "Running: coagulation factor V --> output type SequenceVariant\n",
      "Running: coagulation factor V --> output type ChemicalSubstance\n",
      "Running: coagulation factor V --> output type Disease\n",
      "Running: coagulation factor V --> output type MolecularActivity\n",
      "Running: coagulation factor V --> output type BiologicalProcess\n",
      "Running: coagulation factor V --> output type CellularComponent\n",
      "Running: coagulation factor V --> output type Pathway\n",
      "Running: coagulation factor V --> output type AnatomicalEntity\n",
      "Running: coagulation factor V --> output type PhenotypicFeature\n",
      "Running: small RNA binding exonuclease protection factor La --> output type Gene\n",
      "Running: small RNA binding exonuclease protection factor La --> output type SequenceVariant\n",
      "Running: small RNA binding exonuclease protection factor La --> output type ChemicalSubstance\n",
      "Running: small RNA binding exonuclease protection factor La --> output type Disease\n",
      "Running: small RNA binding exonuclease protection factor La --> output type MolecularActivity\n",
      "Running: small RNA binding exonuclease protection factor La --> output type BiologicalProcess\n",
      "Running: small RNA binding exonuclease protection factor La --> output type CellularComponent\n",
      "Running: small RNA binding exonuclease protection factor La --> output type Pathway\n",
      "Running: small RNA binding exonuclease protection factor La --> output type AnatomicalEntity\n",
      "Running: small RNA binding exonuclease protection factor La --> output type PhenotypicFeature\n",
      "Running: protein S --> output type Gene\n",
      "Running: protein S --> output type SequenceVariant\n",
      "Running: protein S --> output type ChemicalSubstance\n",
      "Running: protein S --> output type Disease\n",
      "Running: protein S --> output type MolecularActivity\n",
      "Running: protein S --> output type BiologicalProcess\n",
      "Running: protein S --> output type CellularComponent\n",
      "Running: protein S --> output type Pathway\n",
      "Running: protein S --> output type AnatomicalEntity\n",
      "Running: protein S --> output type PhenotypicFeature\n",
      "Running: CD82 molecule --> output type Gene\n",
      "Running: CD82 molecule --> output type SequenceVariant\n",
      "Running: CD82 molecule --> output type ChemicalSubstance\n",
      "Running: CD82 molecule --> output type Disease\n",
      "Running: CD82 molecule --> output type MolecularActivity\n",
      "Running: CD82 molecule --> output type BiologicalProcess\n",
      "Running: CD82 molecule --> output type CellularComponent\n",
      "Running: CD82 molecule --> output type Pathway\n",
      "Running: CD82 molecule --> output type AnatomicalEntity\n",
      "Running: CD82 molecule --> output type PhenotypicFeature\n",
      "Running: kallikrein related peptidase 3 --> output type Gene\n",
      "Running: kallikrein related peptidase 3 --> output type SequenceVariant\n",
      "Running: kallikrein related peptidase 3 --> output type ChemicalSubstance\n",
      "Running: kallikrein related peptidase 3 --> output type Disease\n",
      "Running: kallikrein related peptidase 3 --> output type MolecularActivity\n",
      "Running: kallikrein related peptidase 3 --> output type BiologicalProcess\n",
      "Running: kallikrein related peptidase 3 --> output type CellularComponent\n",
      "Running: kallikrein related peptidase 3 --> output type Pathway\n",
      "Running: kallikrein related peptidase 3 --> output type AnatomicalEntity\n",
      "Running: kallikrein related peptidase 3 --> output type PhenotypicFeature\n",
      "Running: serpin family E member 1 --> output type Gene\n",
      "Running: serpin family E member 1 --> output type SequenceVariant\n",
      "Running: serpin family E member 1 --> output type ChemicalSubstance\n",
      "Running: serpin family E member 1 --> output type Disease\n",
      "Running: serpin family E member 1 --> output type MolecularActivity\n",
      "Running: serpin family E member 1 --> output type BiologicalProcess\n",
      "Running: serpin family E member 1 --> output type CellularComponent\n",
      "Running: serpin family E member 1 --> output type Pathway\n",
      "Running: serpin family E member 1 --> output type AnatomicalEntity\n",
      "Running: serpin family E member 1 --> output type PhenotypicFeature\n",
      "Running: serpin family B member 2 --> output type Gene\n",
      "Running: serpin family B member 2 --> output type SequenceVariant\n",
      "Running: serpin family B member 2 --> output type ChemicalSubstance\n",
      "Running: serpin family B member 2 --> output type Disease\n",
      "Running: serpin family B member 2 --> output type MolecularActivity\n",
      "Running: serpin family B member 2 --> output type BiologicalProcess\n",
      "Running: serpin family B member 2 --> output type CellularComponent\n",
      "Running: serpin family B member 2 --> output type Pathway\n",
      "Running: serpin family B member 2 --> output type AnatomicalEntity\n",
      "Running: serpin family B member 2 --> output type PhenotypicFeature\n",
      "Running: Janus kinase 2 --> output type Gene\n",
      "Running: Janus kinase 2 --> output type SequenceVariant\n",
      "Running: Janus kinase 2 --> output type ChemicalSubstance\n",
      "Running: Janus kinase 2 --> output type Disease\n",
      "Running: Janus kinase 2 --> output type MolecularActivity\n",
      "Running: Janus kinase 2 --> output type BiologicalProcess\n",
      "Running: Janus kinase 2 --> output type CellularComponent\n",
      "Running: Janus kinase 2 --> output type Pathway\n",
      "Running: Janus kinase 2 --> output type AnatomicalEntity\n",
      "Running: Janus kinase 2 --> output type PhenotypicFeature\n",
      "Running: twist family bHLH transcription factor 1 --> output type Gene\n",
      "Running: twist family bHLH transcription factor 1 --> output type SequenceVariant\n",
      "Running: twist family bHLH transcription factor 1 --> output type ChemicalSubstance\n",
      "Running: twist family bHLH transcription factor 1 --> output type Disease\n",
      "Running: twist family bHLH transcription factor 1 --> output type MolecularActivity\n",
      "Running: twist family bHLH transcription factor 1 --> output type BiologicalProcess\n",
      "Running: twist family bHLH transcription factor 1 --> output type CellularComponent\n",
      "Running: twist family bHLH transcription factor 1 --> output type Pathway\n",
      "Running: twist family bHLH transcription factor 1 --> output type AnatomicalEntity\n",
      "Running: twist family bHLH transcription factor 1 --> output type PhenotypicFeature\n",
      "Running: tubulin beta 4B class IVb --> output type Gene\n",
      "Running: tubulin beta 4B class IVb --> output type SequenceVariant\n",
      "Running: tubulin beta 4B class IVb --> output type ChemicalSubstance\n",
      "Running: tubulin beta 4B class IVb --> output type Disease\n",
      "Running: tubulin beta 4B class IVb --> output type MolecularActivity\n",
      "Running: tubulin beta 4B class IVb --> output type BiologicalProcess\n",
      "Running: tubulin beta 4B class IVb --> output type CellularComponent\n",
      "Running: tubulin beta 4B class IVb --> output type Pathway\n",
      "Running: tubulin beta 4B class IVb --> output type AnatomicalEntity\n",
      "Running: tubulin beta 4B class IVb --> output type PhenotypicFeature\n",
      "Running: neuronal differentiation 1 --> output type Gene\n",
      "Running: neuronal differentiation 1 --> output type SequenceVariant\n",
      "Running: neuronal differentiation 1 --> output type ChemicalSubstance\n",
      "Running: neuronal differentiation 1 --> output type Disease\n",
      "Running: neuronal differentiation 1 --> output type MolecularActivity\n",
      "Running: neuronal differentiation 1 --> output type BiologicalProcess\n",
      "Running: neuronal differentiation 1 --> output type CellularComponent\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running: neuronal differentiation 1 --> output type Pathway\n",
      "Running: neuronal differentiation 1 --> output type AnatomicalEntity\n",
      "Running: neuronal differentiation 1 --> output type PhenotypicFeature\n",
      "Running: ceruloplasmin --> output type Gene\n",
      "Running: ceruloplasmin --> output type SequenceVariant\n",
      "Running: ceruloplasmin --> output type ChemicalSubstance\n",
      "Running: ceruloplasmin --> output type Disease\n",
      "Running: ceruloplasmin --> output type MolecularActivity\n",
      "Running: ceruloplasmin --> output type BiologicalProcess\n",
      "Running: ceruloplasmin --> output type CellularComponent\n",
      "Running: ceruloplasmin --> output type Pathway\n",
      "Running: ceruloplasmin --> output type AnatomicalEntity\n",
      "Running: ceruloplasmin --> output type PhenotypicFeature\n",
      "Running: glucose-6-phosphate isomerase --> output type Gene\n",
      "Running: glucose-6-phosphate isomerase --> output type SequenceVariant\n",
      "Running: glucose-6-phosphate isomerase --> output type ChemicalSubstance\n",
      "Running: glucose-6-phosphate isomerase --> output type Disease\n",
      "Running: glucose-6-phosphate isomerase --> output type MolecularActivity\n",
      "Running: glucose-6-phosphate isomerase --> output type BiologicalProcess\n",
      "Running: glucose-6-phosphate isomerase --> output type CellularComponent\n",
      "Running: glucose-6-phosphate isomerase --> output type Pathway\n",
      "Running: glucose-6-phosphate isomerase --> output type AnatomicalEntity\n",
      "Running: glucose-6-phosphate isomerase --> output type PhenotypicFeature\n",
      "Running: coagulation factor VIII --> output type Gene\n",
      "Running: coagulation factor VIII --> output type SequenceVariant\n",
      "Running: coagulation factor VIII --> output type ChemicalSubstance\n",
      "Running: coagulation factor VIII --> output type Disease\n",
      "Running: coagulation factor VIII --> output type MolecularActivity\n",
      "Running: coagulation factor VIII --> output type BiologicalProcess\n",
      "Running: coagulation factor VIII --> output type CellularComponent\n",
      "Running: coagulation factor VIII --> output type Pathway\n",
      "Running: coagulation factor VIII --> output type AnatomicalEntity\n",
      "Running: coagulation factor VIII --> output type PhenotypicFeature\n",
      "Running: plasminogen activator, tissue type --> output type Gene\n",
      "Running: plasminogen activator, tissue type --> output type SequenceVariant\n",
      "Running: plasminogen activator, tissue type --> output type ChemicalSubstance\n",
      "Running: plasminogen activator, tissue type --> output type Disease\n",
      "Running: plasminogen activator, tissue type --> output type MolecularActivity\n",
      "Running: plasminogen activator, tissue type --> output type BiologicalProcess\n",
      "Running: plasminogen activator, tissue type --> output type CellularComponent\n",
      "Running: plasminogen activator, tissue type --> output type Pathway\n",
      "Running: plasminogen activator, tissue type --> output type AnatomicalEntity\n",
      "Running: plasminogen activator, tissue type --> output type PhenotypicFeature\n",
      "Running: plasminogen --> output type Gene\n",
      "Running: plasminogen --> output type SequenceVariant\n",
      "Running: plasminogen --> output type ChemicalSubstance\n",
      "Running: plasminogen --> output type Disease\n",
      "Running: plasminogen --> output type MolecularActivity\n",
      "Running: plasminogen --> output type BiologicalProcess\n",
      "Running: plasminogen --> output type CellularComponent\n",
      "Running: plasminogen --> output type Pathway\n",
      "Running: plasminogen --> output type AnatomicalEntity\n",
      "Running: plasminogen --> output type PhenotypicFeature\n",
      "Running: protein C, inactivator of coagulation factors Va and VIIIa --> output type Gene\n",
      "Running: protein C, inactivator of coagulation factors Va and VIIIa --> output type SequenceVariant\n",
      "Running: protein C, inactivator of coagulation factors Va and VIIIa --> output type ChemicalSubstance\n",
      "Running: protein C, inactivator of coagulation factors Va and VIIIa --> output type Disease\n",
      "Running: protein C, inactivator of coagulation factors Va and VIIIa --> output type MolecularActivity\n",
      "Running: protein C, inactivator of coagulation factors Va and VIIIa --> output type BiologicalProcess\n",
      "Running: protein C, inactivator of coagulation factors Va and VIIIa --> output type CellularComponent\n",
      "Running: protein C, inactivator of coagulation factors Va and VIIIa --> output type Pathway\n",
      "Running: protein C, inactivator of coagulation factors Va and VIIIa --> output type AnatomicalEntity\n",
      "Running: protein C, inactivator of coagulation factors Va and VIIIa --> output type PhenotypicFeature\n",
      "Running: lipoprotein(a) --> output type Gene\n",
      "Running: lipoprotein(a) --> output type SequenceVariant\n",
      "Running: lipoprotein(a) --> output type ChemicalSubstance\n",
      "Running: lipoprotein(a) --> output type Disease\n",
      "Running: lipoprotein(a) --> output type MolecularActivity\n",
      "Running: lipoprotein(a) --> output type BiologicalProcess\n",
      "Running: lipoprotein(a) --> output type CellularComponent\n",
      "Running: lipoprotein(a) --> output type Pathway\n",
      "Running: lipoprotein(a) --> output type AnatomicalEntity\n",
      "Running: lipoprotein(a) --> output type PhenotypicFeature\n",
      "Running: AKT serine/threonine kinase 1 --> output type Gene\n",
      "Running: AKT serine/threonine kinase 1 --> output type SequenceVariant\n",
      "Running: AKT serine/threonine kinase 1 --> output type ChemicalSubstance\n",
      "Running: AKT serine/threonine kinase 1 --> output type Disease\n",
      "Running: AKT serine/threonine kinase 1 --> output type MolecularActivity\n",
      "Running: AKT serine/threonine kinase 1 --> output type BiologicalProcess\n",
      "Running: AKT serine/threonine kinase 1 --> output type CellularComponent\n",
      "Running: AKT serine/threonine kinase 1 --> output type Pathway\n",
      "Running: AKT serine/threonine kinase 1 --> output type AnatomicalEntity\n",
      "Running: AKT serine/threonine kinase 1 --> output type PhenotypicFeature\n",
      "Running: transforming growth factor beta 1 --> output type Gene\n",
      "Running: transforming growth factor beta 1 --> output type SequenceVariant\n",
      "Running: transforming growth factor beta 1 --> output type ChemicalSubstance\n",
      "Running: transforming growth factor beta 1 --> output type Disease\n",
      "Running: transforming growth factor beta 1 --> output type MolecularActivity\n",
      "Running: transforming growth factor beta 1 --> output type BiologicalProcess\n",
      "Running: transforming growth factor beta 1 --> output type CellularComponent\n",
      "Running: transforming growth factor beta 1 --> output type Pathway\n",
      "Running: transforming growth factor beta 1 --> output type AnatomicalEntity\n",
      "Running: transforming growth factor beta 1 --> output type PhenotypicFeature\n",
      "Running: epidermal growth factor receptor --> output type Gene\n",
      "Running: epidermal growth factor receptor --> output type SequenceVariant\n",
      "Running: epidermal growth factor receptor --> output type ChemicalSubstance\n",
      "Running: epidermal growth factor receptor --> output type Disease\n",
      "Running: epidermal growth factor receptor --> output type MolecularActivity\n",
      "Running: epidermal growth factor receptor --> output type BiologicalProcess\n",
      "Running: epidermal growth factor receptor --> output type CellularComponent\n",
      "Running: epidermal growth factor receptor --> output type Pathway\n",
      "Running: epidermal growth factor receptor --> output type AnatomicalEntity\n",
      "Running: epidermal growth factor receptor --> output type PhenotypicFeature\n",
      "Running: matrix metallopeptidase 2 --> output type Gene\n",
      "Running: matrix metallopeptidase 2 --> output type SequenceVariant\n",
      "Running: matrix metallopeptidase 2 --> output type ChemicalSubstance\n",
      "Running: matrix metallopeptidase 2 --> output type Disease\n",
      "Running: matrix metallopeptidase 2 --> output type MolecularActivity\n",
      "Running: matrix metallopeptidase 2 --> output type BiologicalProcess\n",
      "Running: matrix metallopeptidase 2 --> output type CellularComponent\n",
      "Running: matrix metallopeptidase 2 --> output type Pathway\n",
      "Running: matrix metallopeptidase 2 --> output type AnatomicalEntity\n",
      "Running: matrix metallopeptidase 2 --> output type PhenotypicFeature\n",
      "Running: methylenetetrahydrofolate reductase --> output type Gene\n",
      "Running: methylenetetrahydrofolate reductase --> output type SequenceVariant\n",
      "Running: methylenetetrahydrofolate reductase --> output type ChemicalSubstance\n",
      "Running: methylenetetrahydrofolate reductase --> output type Disease\n",
      "Running: methylenetetrahydrofolate reductase --> output type MolecularActivity\n",
      "Running: methylenetetrahydrofolate reductase --> output type BiologicalProcess\n",
      "Running: methylenetetrahydrofolate reductase --> output type CellularComponent\n",
      "Running: methylenetetrahydrofolate reductase --> output type Pathway\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running: methylenetetrahydrofolate reductase --> output type AnatomicalEntity\n",
      "Running: methylenetetrahydrofolate reductase --> output type PhenotypicFeature\n",
      "Running: interferon alpha 1 --> output type Gene\n",
      "Running: interferon alpha 1 --> output type SequenceVariant\n",
      "Running: interferon alpha 1 --> output type ChemicalSubstance\n",
      "Running: interferon alpha 1 --> output type Disease\n",
      "Running: interferon alpha 1 --> output type MolecularActivity\n",
      "Running: interferon alpha 1 --> output type BiologicalProcess\n",
      "Running: interferon alpha 1 --> output type CellularComponent\n",
      "Running: interferon alpha 1 --> output type Pathway\n",
      "Running: interferon alpha 1 --> output type AnatomicalEntity\n",
      "Running: interferon alpha 1 --> output type PhenotypicFeature\n",
      "Running: parathyroid hormone --> output type Gene\n",
      "Running: parathyroid hormone --> output type SequenceVariant\n",
      "Running: parathyroid hormone --> output type ChemicalSubstance\n",
      "Running: parathyroid hormone --> output type Disease\n",
      "Running: parathyroid hormone --> output type MolecularActivity\n",
      "Running: parathyroid hormone --> output type BiologicalProcess\n",
      "Running: parathyroid hormone --> output type CellularComponent\n",
      "Running: parathyroid hormone --> output type Pathway\n",
      "Running: parathyroid hormone --> output type AnatomicalEntity\n",
      "Running: parathyroid hormone --> output type PhenotypicFeature\n",
      "Running: von Willebrand factor --> output type Gene\n",
      "Running: von Willebrand factor --> output type SequenceVariant\n",
      "Running: von Willebrand factor --> output type ChemicalSubstance\n",
      "Running: von Willebrand factor --> output type Disease\n",
      "Running: von Willebrand factor --> output type MolecularActivity\n",
      "Running: von Willebrand factor --> output type BiologicalProcess\n",
      "Running: von Willebrand factor --> output type CellularComponent\n",
      "Running: von Willebrand factor --> output type Pathway\n",
      "Running: von Willebrand factor --> output type AnatomicalEntity\n",
      "Running: von Willebrand factor --> output type PhenotypicFeature\n",
      "Running: cytokine inducible SH2 containing protein --> output type Gene\n",
      "Running: cytokine inducible SH2 containing protein --> output type SequenceVariant\n",
      "Running: cytokine inducible SH2 containing protein --> output type ChemicalSubstance\n",
      "Running: cytokine inducible SH2 containing protein --> output type Disease\n",
      "Running: cytokine inducible SH2 containing protein --> output type MolecularActivity\n",
      "Running: cytokine inducible SH2 containing protein --> output type BiologicalProcess\n",
      "Running: cytokine inducible SH2 containing protein --> output type CellularComponent\n",
      "Running: cytokine inducible SH2 containing protein --> output type Pathway\n",
      "Running: cytokine inducible SH2 containing protein --> output type AnatomicalEntity\n",
      "Running: cytokine inducible SH2 containing protein --> output type PhenotypicFeature\n",
      "Running: angiotensin I converting enzyme --> output type Gene\n",
      "Running: angiotensin I converting enzyme --> output type SequenceVariant\n",
      "Running: angiotensin I converting enzyme --> output type ChemicalSubstance\n",
      "Running: angiotensin I converting enzyme --> output type Disease\n",
      "Running: angiotensin I converting enzyme --> output type MolecularActivity\n",
      "Running: angiotensin I converting enzyme --> output type BiologicalProcess\n",
      "Running: angiotensin I converting enzyme --> output type CellularComponent\n",
      "Running: angiotensin I converting enzyme --> output type Pathway\n",
      "Running: angiotensin I converting enzyme --> output type AnatomicalEntity\n",
      "Running: angiotensin I converting enzyme --> output type PhenotypicFeature\n",
      "Running: G protein-coupled receptor kinase 2 --> output type Gene\n",
      "Running: G protein-coupled receptor kinase 2 --> output type SequenceVariant\n",
      "Running: G protein-coupled receptor kinase 2 --> output type ChemicalSubstance\n",
      "Running: G protein-coupled receptor kinase 2 --> output type Disease\n",
      "Running: G protein-coupled receptor kinase 2 --> output type MolecularActivity\n",
      "Running: G protein-coupled receptor kinase 2 --> output type BiologicalProcess\n",
      "Running: G protein-coupled receptor kinase 2 --> output type CellularComponent\n",
      "Running: G protein-coupled receptor kinase 2 --> output type Pathway\n",
      "Running: G protein-coupled receptor kinase 2 --> output type AnatomicalEntity\n",
      "Running: G protein-coupled receptor kinase 2 --> output type PhenotypicFeature\n",
      "Running: KIT proto-oncogene, receptor tyrosine kinase --> output type Gene\n",
      "Running: KIT proto-oncogene, receptor tyrosine kinase --> output type SequenceVariant\n",
      "Running: KIT proto-oncogene, receptor tyrosine kinase --> output type ChemicalSubstance\n",
      "Running: KIT proto-oncogene, receptor tyrosine kinase --> output type Disease\n",
      "Running: KIT proto-oncogene, receptor tyrosine kinase --> output type MolecularActivity\n",
      "Running: KIT proto-oncogene, receptor tyrosine kinase --> output type BiologicalProcess\n",
      "Running: KIT proto-oncogene, receptor tyrosine kinase --> output type CellularComponent\n",
      "Running: KIT proto-oncogene, receptor tyrosine kinase --> output type Pathway\n",
      "Running: KIT proto-oncogene, receptor tyrosine kinase --> output type AnatomicalEntity\n",
      "Running: KIT proto-oncogene, receptor tyrosine kinase --> output type PhenotypicFeature\n",
      "Running: phosphatase and tensin homolog --> output type Gene\n",
      "Running: phosphatase and tensin homolog --> output type SequenceVariant\n",
      "Running: phosphatase and tensin homolog --> output type ChemicalSubstance\n",
      "Running: phosphatase and tensin homolog --> output type Disease\n",
      "Running: phosphatase and tensin homolog --> output type MolecularActivity\n",
      "Running: phosphatase and tensin homolog --> output type BiologicalProcess\n",
      "Running: phosphatase and tensin homolog --> output type CellularComponent\n",
      "Running: phosphatase and tensin homolog --> output type Pathway\n",
      "Running: phosphatase and tensin homolog --> output type AnatomicalEntity\n",
      "Running: phosphatase and tensin homolog --> output type PhenotypicFeature\n",
      "Running: ADAM metallopeptidase with thrombospondin type 1 motif 13 --> output type Gene\n",
      "Running: ADAM metallopeptidase with thrombospondin type 1 motif 13 --> output type SequenceVariant\n",
      "Running: ADAM metallopeptidase with thrombospondin type 1 motif 13 --> output type ChemicalSubstance\n",
      "Running: ADAM metallopeptidase with thrombospondin type 1 motif 13 --> output type Disease\n",
      "Running: ADAM metallopeptidase with thrombospondin type 1 motif 13 --> output type MolecularActivity\n",
      "Running: ADAM metallopeptidase with thrombospondin type 1 motif 13 --> output type BiologicalProcess\n",
      "Running: ADAM metallopeptidase with thrombospondin type 1 motif 13 --> output type CellularComponent\n",
      "Running: ADAM metallopeptidase with thrombospondin type 1 motif 13 --> output type Pathway\n",
      "Running: ADAM metallopeptidase with thrombospondin type 1 motif 13 --> output type AnatomicalEntity\n",
      "Running: ADAM metallopeptidase with thrombospondin type 1 motif 13 --> output type PhenotypicFeature\n",
      "Running: tyrosine aminotransferase --> output type Gene\n",
      "Running: tyrosine aminotransferase --> output type SequenceVariant\n",
      "Running: tyrosine aminotransferase --> output type ChemicalSubstance\n",
      "Running: tyrosine aminotransferase --> output type Disease\n",
      "Running: tyrosine aminotransferase --> output type MolecularActivity\n",
      "Running: tyrosine aminotransferase --> output type BiologicalProcess\n",
      "Running: tyrosine aminotransferase --> output type CellularComponent\n",
      "Running: tyrosine aminotransferase --> output type Pathway\n",
      "Running: tyrosine aminotransferase --> output type AnatomicalEntity\n",
      "Running: tyrosine aminotransferase --> output type PhenotypicFeature\n",
      "Running: transferrin --> output type Gene\n",
      "Running: transferrin --> output type SequenceVariant\n",
      "Running: transferrin --> output type ChemicalSubstance\n",
      "Running: transferrin --> output type Disease\n",
      "Running: transferrin --> output type MolecularActivity\n",
      "Running: transferrin --> output type BiologicalProcess\n",
      "Running: transferrin --> output type CellularComponent\n",
      "Running: transferrin --> output type Pathway\n",
      "Running: transferrin --> output type AnatomicalEntity\n",
      "Running: transferrin --> output type PhenotypicFeature\n",
      "{'CDSN': 960, 'TFPI': 1275, 'F5': 1001, 'SSB': 1380, 'PROS1': 705, 'CD82': 1676, 'KLK3': 2156, 'SERPINE1': 1935, 'SERPINB2': 1297, 'JAK2': 2975, 'TWIST1': 3211, 'TUBB4B': 2542, 'NEUROD1': 2477, 'CP': 2682, 'GPI': 3194, 'F8': 1148, 'PLAT': 2526, 'PLG': 1275, 'PROC': 821, 'LPA': 1428, 'AKT1': 16400, 'TGFB1': 9079, 'EGFR': 9004, 'MMP2': 7392, 'MTHFR': 1639, 'IFNA1': 7673, 'PTH': 4705, 'VWF': 1607, 'CISH': 6348, 'ACE': 2842, 'GRK2': 4431, 'KIT': 4843, 'PTEN': 5807, 'ADAMTS13': 760, 'TAT': 2926, 'TF': 2821}\n"
     ]
    }
   ],
   "source": [
    "# get edges out count from genes to any node type to get rough estimate of how well researched a gene is\n",
    "relevant_gene_inputs = []\n",
    "\n",
    "for gene_input in gene_inputs: \n",
    "    if(gene_input['SYMBOL'] in relevant_top_genes_list):\n",
    "        relevant_gene_inputs.append(gene_input)\n",
    "        \n",
    "all_gene_edges_out = predict_many(relevant_gene_inputs, ALL_NODE_TYPES)\n",
    "edges_out_genes_list = list(all_gene_edges_out[\"input\"])\n",
    "gene_edges_out = {x:edges_out_genes_list.count(x) for x in edges_out_genes_list}\n",
    "print(gene_edges_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Stored 'gene_edges_out' (dict)\n"
     ]
    }
   ],
   "source": [
    "%store gene_edges_out\n",
    "%store -r gene_edges_out"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 7. Assemble Genes related to both Disease and Disease Symptoms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# assemble final dictionary that includes all metrics for gene's connections to disease (direct and through one node)\n",
    "# as well as to disease symptoms\n",
    "final_dict = {}\n",
    "\n",
    "for x in relevant_top_genes_list:\n",
    "    symptoms_to_hpids = [];\n",
    "    for symptom in symptoms_results[x]['related_symptoms']: \n",
    "        symptoms_to_hpids.append(symptom_to_hpid_dict[symptom])\n",
    "    unique_symptoms = list(dict.fromkeys(symptoms_to_hpids))\n",
    "    final_dict[x] = {\n",
    "        \"disease_to_gene_occurrences\" : disease_to_gene_results[x]['gene_count'] if x in disease_to_gene_results else 0,\n",
    "        \"disease_to_gene_pub_counts\" : len(disease_to_gene_results[x]['publications']) if x in disease_to_gene_results else 0,\n",
    "        \"disease_to_int_to_gen_occurrences\" : disease_to_all_nodes_to_genes_results[x]['gene_count'] if x in disease_to_all_nodes_to_genes_results else 0,\n",
    "        \"disease_to_int_to_gene_pubs\" : len(disease_to_all_nodes_to_genes_results[x]['publications']) if x in disease_to_all_nodes_to_genes_results else 0,\n",
    "        \"disease_symtpoms_gene_related_to\" : symptoms_results[x]['related_symptoms'],\n",
    "        \"disease_symtpoms_gene_related_to_count\" : len(symptoms_results[x]['related_symptoms']),\n",
    "        \"unique_symptoms_count\": len(unique_symptoms),\n",
    "        \"gene_to_symptoms_pub_counts\" : len(symptoms_results[x]['publications']),\n",
    "        \"gene_edges_out\": gene_edges_out[x]\n",
    "    }\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>disease_to_gene_occurrences</th>\n",
       "      <th>disease_to_gene_pub_counts</th>\n",
       "      <th>disease_to_int_to_gen_occurrences</th>\n",
       "      <th>disease_to_int_to_gene_pubs</th>\n",
       "      <th>disease_symtpoms_gene_related_to</th>\n",
       "      <th>disease_symtpoms_gene_related_to_count</th>\n",
       "      <th>unique_symptoms_count</th>\n",
       "      <th>gene_to_symptoms_pub_counts</th>\n",
       "      <th>gene_edges_out</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CDSN</th>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>227</td>\n",
       "      <td>560</td>\n",
       "      <td>[ACTIVATED PROTEIN C RESISTANCE, ACTIVATED PRO...</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TFPI</th>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>176</td>\n",
       "      <td>322</td>\n",
       "      <td>[ACTIVATED PROTEIN C RESISTANCE, ACTIVATED PRO...</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>F5</th>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>64</td>\n",
       "      <td>87</td>\n",
       "      <td>[ABNORMAL PARTIAL THROMBOPLASTIN TIME, BLOOD C...</td>\n",
       "      <td>9</td>\n",
       "      <td>5</td>\n",
       "      <td>9</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SSB</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>104</td>\n",
       "      <td>208</td>\n",
       "      <td>[ACTIVATED PROTEIN C RESISTANCE, ACTIVATED PRO...</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PROS1</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>73</td>\n",
       "      <td>126</td>\n",
       "      <td>[BLOOD HYPERVISCOSITY, BLOOD CLOT IN A DEEP VE...</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "      <td>705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD82</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>41</td>\n",
       "      <td>80</td>\n",
       "      <td>[ACTIVATED PROTEIN C RESISTANCE]</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KLK3</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>144</td>\n",
       "      <td>483</td>\n",
       "      <td>[DEEP VEIN THROMBOSIS, ACTIVATED PROTEIN C RES...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SERPINE1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>399</td>\n",
       "      <td>[DEEP VEIN THROMBOSIS, ACTIVATED PROTEIN C RES...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SERPINB2</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>170</td>\n",
       "      <td>332</td>\n",
       "      <td>[ACTIVATED PROTEIN C RESISTANCE]</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>JAK2</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>101</td>\n",
       "      <td>202</td>\n",
       "      <td>[DEEP VEIN THROMBOSIS, DEEP VEIN THROMBOSIS, A...</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TWIST1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>89</td>\n",
       "      <td>259</td>\n",
       "      <td>[AUTOSOMAL DOMINANT, AUTOSOMAL DOMINANT, ACTIV...</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>3211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TUBB4B</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>148</td>\n",
       "      <td>282</td>\n",
       "      <td>[ACTIVATED PROTEIN C RESISTANCE]</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEUROD1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>148</td>\n",
       "      <td>289</td>\n",
       "      <td>[ACTIVATED PROTEIN C RESISTANCE]</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CP</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>121</td>\n",
       "      <td>220</td>\n",
       "      <td>[ACTIVATED PROTEIN C RESISTANCE]</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GPI</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>171</td>\n",
       "      <td>627</td>\n",
       "      <td>[ACTIVATED PROTEIN C RESISTANCE, DEEP VEIN THR...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>3194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>F8</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>127</td>\n",
       "      <td>266</td>\n",
       "      <td>[ABNORMAL PARTIAL THROMBOPLASTIN TIME, DEEP VE...</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>1148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PLAT</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>176</td>\n",
       "      <td>591</td>\n",
       "      <td>[BLOOD HYPERVISCOSITY, DEEP VEIN THROMBOSIS, D...</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PLG</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>94</td>\n",
       "      <td>121</td>\n",
       "      <td>[DEEP VEIN THROMBOSIS, DEEP VEIN THROMBOSIS, A...</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PROC</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>77</td>\n",
       "      <td>[BLOOD CLOT IN A DEEP VEIN, BLOOD HYPERVISCOSI...</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LPA</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>115</td>\n",
       "      <td>304</td>\n",
       "      <td>[DEEP VEIN THROMBOSIS, DEEP VEIN THROMBOSIS, A...</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKT1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>313</td>\n",
       "      <td>1341</td>\n",
       "      <td>[BLOOD CLOT IN A DEEP VEIN]</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>16400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TGFB1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>259</td>\n",
       "      <td>950</td>\n",
       "      <td>[AUTOSOMAL DOMINANT, DEEP VEIN THROMBOSIS]</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>9079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EGFR</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>255</td>\n",
       "      <td>1737</td>\n",
       "      <td>[DEEP VEIN THROMBOSIS]</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>9004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MMP2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>242</td>\n",
       "      <td>532</td>\n",
       "      <td>[DEEP VEIN THROMBOSIS, DEEP VEIN THROMBOSIS]</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>7392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTHFR</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>216</td>\n",
       "      <td>441</td>\n",
       "      <td>[ABNORMAL PARTIAL THROMBOPLASTIN TIME, DEEP VE...</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>1639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IFNA1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>205</td>\n",
       "      <td>909</td>\n",
       "      <td>[DEEP VEIN THROMBOSIS]</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>7673</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTH</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>198</td>\n",
       "      <td>572</td>\n",
       "      <td>[AUTOSOMAL DOMINANT]</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>4705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VWF</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>188</td>\n",
       "      <td>260</td>\n",
       "      <td>[DEEP VEIN THROMBOSIS, DEEP VEIN THROMBOSIS]</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CISH</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>164</td>\n",
       "      <td>612</td>\n",
       "      <td>[DEEP VEIN THROMBOSIS]</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACE</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>159</td>\n",
       "      <td>274</td>\n",
       "      <td>[DEEP VEIN THROMBOSIS]</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GRK2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>151</td>\n",
       "      <td>450</td>\n",
       "      <td>[DEEP VEIN THROMBOSIS]</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>4431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KIT</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>143</td>\n",
       "      <td>379</td>\n",
       "      <td>[AUTOSOMAL DOMINANT]</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>4843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTEN</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>143</td>\n",
       "      <td>628</td>\n",
       "      <td>[AUTOSOMAL DOMINANT, AUTOSOMAL DOMINANT, AUTOS...</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>5807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ADAMTS13</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>135</td>\n",
       "      <td>203</td>\n",
       "      <td>[DEEP VEIN THROMBOSIS, DEEP VEIN THROMBOSIS]</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TAT</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>132</td>\n",
       "      <td>246</td>\n",
       "      <td>[DEEP VEIN THROMBOSIS]</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TF</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>121</td>\n",
       "      <td>223</td>\n",
       "      <td>[DEEP VEIN THROMBOSIS]</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2821</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         disease_to_gene_occurrences disease_to_gene_pub_counts  \\\n",
       "CDSN                               4                          7   \n",
       "TFPI                               3                          1   \n",
       "F5                                 3                          0   \n",
       "SSB                                2                          2   \n",
       "PROS1                              2                          0   \n",
       "CD82                               1                          1   \n",
       "KLK3                               1                          1   \n",
       "SERPINE1                           1                          1   \n",
       "SERPINB2                           1                          1   \n",
       "JAK2                               1                          1   \n",
       "TWIST1                             1                          1   \n",
       "TUBB4B                             1                          1   \n",
       "NEUROD1                            1                          1   \n",
       "CP                                 1                          1   \n",
       "GPI                                1                          1   \n",
       "F8                                 1                          0   \n",
       "PLAT                               1                          0   \n",
       "PLG                                1                          0   \n",
       "PROC                               1                          0   \n",
       "LPA                                1                          0   \n",
       "AKT1                               0                          0   \n",
       "TGFB1                              0                          0   \n",
       "EGFR                               0                          0   \n",
       "MMP2                               0                          0   \n",
       "MTHFR                              0                          0   \n",
       "IFNA1                              0                          0   \n",
       "PTH                                0                          0   \n",
       "VWF                                0                          0   \n",
       "CISH                               0                          0   \n",
       "ACE                                0                          0   \n",
       "GRK2                               0                          0   \n",
       "KIT                                0                          0   \n",
       "PTEN                               0                          0   \n",
       "ADAMTS13                           0                          0   \n",
       "TAT                                0                          0   \n",
       "TF                                 0                          0   \n",
       "\n",
       "         disease_to_int_to_gen_occurrences disease_to_int_to_gene_pubs  \\\n",
       "CDSN                                   227                         560   \n",
       "TFPI                                   176                         322   \n",
       "F5                                      64                          87   \n",
       "SSB                                    104                         208   \n",
       "PROS1                                   73                         126   \n",
       "CD82                                    41                          80   \n",
       "KLK3                                   144                         483   \n",
       "SERPINE1                               217                         399   \n",
       "SERPINB2                               170                         332   \n",
       "JAK2                                   101                         202   \n",
       "TWIST1                                  89                         259   \n",
       "TUBB4B                                 148                         282   \n",
       "NEUROD1                                148                         289   \n",
       "CP                                     121                         220   \n",
       "GPI                                    171                         627   \n",
       "F8                                     127                         266   \n",
       "PLAT                                   176                         591   \n",
       "PLG                                     94                         121   \n",
       "PROC                                    56                          77   \n",
       "LPA                                    115                         304   \n",
       "AKT1                                   313                        1341   \n",
       "TGFB1                                  259                         950   \n",
       "EGFR                                   255                        1737   \n",
       "MMP2                                   242                         532   \n",
       "MTHFR                                  216                         441   \n",
       "IFNA1                                  205                         909   \n",
       "PTH                                    198                         572   \n",
       "VWF                                    188                         260   \n",
       "CISH                                   164                         612   \n",
       "ACE                                    159                         274   \n",
       "GRK2                                   151                         450   \n",
       "KIT                                    143                         379   \n",
       "PTEN                                   143                         628   \n",
       "ADAMTS13                               135                         203   \n",
       "TAT                                    132                         246   \n",
       "TF                                     121                         223   \n",
       "\n",
       "                           disease_symtpoms_gene_related_to  \\\n",
       "CDSN      [ACTIVATED PROTEIN C RESISTANCE, ACTIVATED PRO...   \n",
       "TFPI      [ACTIVATED PROTEIN C RESISTANCE, ACTIVATED PRO...   \n",
       "F5        [ABNORMAL PARTIAL THROMBOPLASTIN TIME, BLOOD C...   \n",
       "SSB       [ACTIVATED PROTEIN C RESISTANCE, ACTIVATED PRO...   \n",
       "PROS1     [BLOOD HYPERVISCOSITY, BLOOD CLOT IN A DEEP VE...   \n",
       "CD82                       [ACTIVATED PROTEIN C RESISTANCE]   \n",
       "KLK3      [DEEP VEIN THROMBOSIS, ACTIVATED PROTEIN C RES...   \n",
       "SERPINE1  [DEEP VEIN THROMBOSIS, ACTIVATED PROTEIN C RES...   \n",
       "SERPINB2                   [ACTIVATED PROTEIN C RESISTANCE]   \n",
       "JAK2      [DEEP VEIN THROMBOSIS, DEEP VEIN THROMBOSIS, A...   \n",
       "TWIST1    [AUTOSOMAL DOMINANT, AUTOSOMAL DOMINANT, ACTIV...   \n",
       "TUBB4B                     [ACTIVATED PROTEIN C RESISTANCE]   \n",
       "NEUROD1                    [ACTIVATED PROTEIN C RESISTANCE]   \n",
       "CP                         [ACTIVATED PROTEIN C RESISTANCE]   \n",
       "GPI       [ACTIVATED PROTEIN C RESISTANCE, DEEP VEIN THR...   \n",
       "F8        [ABNORMAL PARTIAL THROMBOPLASTIN TIME, DEEP VE...   \n",
       "PLAT      [BLOOD HYPERVISCOSITY, DEEP VEIN THROMBOSIS, D...   \n",
       "PLG       [DEEP VEIN THROMBOSIS, DEEP VEIN THROMBOSIS, A...   \n",
       "PROC      [BLOOD CLOT IN A DEEP VEIN, BLOOD HYPERVISCOSI...   \n",
       "LPA       [DEEP VEIN THROMBOSIS, DEEP VEIN THROMBOSIS, A...   \n",
       "AKT1                            [BLOOD CLOT IN A DEEP VEIN]   \n",
       "TGFB1            [AUTOSOMAL DOMINANT, DEEP VEIN THROMBOSIS]   \n",
       "EGFR                                 [DEEP VEIN THROMBOSIS]   \n",
       "MMP2           [DEEP VEIN THROMBOSIS, DEEP VEIN THROMBOSIS]   \n",
       "MTHFR     [ABNORMAL PARTIAL THROMBOPLASTIN TIME, DEEP VE...   \n",
       "IFNA1                                [DEEP VEIN THROMBOSIS]   \n",
       "PTH                                    [AUTOSOMAL DOMINANT]   \n",
       "VWF            [DEEP VEIN THROMBOSIS, DEEP VEIN THROMBOSIS]   \n",
       "CISH                                 [DEEP VEIN THROMBOSIS]   \n",
       "ACE                                  [DEEP VEIN THROMBOSIS]   \n",
       "GRK2                                 [DEEP VEIN THROMBOSIS]   \n",
       "KIT                                    [AUTOSOMAL DOMINANT]   \n",
       "PTEN      [AUTOSOMAL DOMINANT, AUTOSOMAL DOMINANT, AUTOS...   \n",
       "ADAMTS13       [DEEP VEIN THROMBOSIS, DEEP VEIN THROMBOSIS]   \n",
       "TAT                                  [DEEP VEIN THROMBOSIS]   \n",
       "TF                                   [DEEP VEIN THROMBOSIS]   \n",
       "\n",
       "         disease_symtpoms_gene_related_to_count unique_symptoms_count  \\\n",
       "CDSN                                          8                     2   \n",
       "TFPI                                          5                     2   \n",
       "F5                                            9                     5   \n",
       "SSB                                           2                     1   \n",
       "PROS1                                         6                     3   \n",
       "CD82                                          1                     1   \n",
       "KLK3                                          2                     2   \n",
       "SERPINE1                                      2                     2   \n",
       "SERPINB2                                      1                     1   \n",
       "JAK2                                          3                     2   \n",
       "TWIST1                                        3                     2   \n",
       "TUBB4B                                        1                     1   \n",
       "NEUROD1                                       1                     1   \n",
       "CP                                            1                     1   \n",
       "GPI                                           2                     2   \n",
       "F8                                            4                     3   \n",
       "PLAT                                          4                     3   \n",
       "PLG                                           3                     2   \n",
       "PROC                                          5                     3   \n",
       "LPA                                           3                     2   \n",
       "AKT1                                          1                     1   \n",
       "TGFB1                                         2                     2   \n",
       "EGFR                                          1                     1   \n",
       "MMP2                                          2                     1   \n",
       "MTHFR                                         5                     2   \n",
       "IFNA1                                         1                     1   \n",
       "PTH                                           1                     1   \n",
       "VWF                                           2                     1   \n",
       "CISH                                          1                     1   \n",
       "ACE                                           1                     1   \n",
       "GRK2                                          1                     1   \n",
       "KIT                                           1                     1   \n",
       "PTEN                                          4                     2   \n",
       "ADAMTS13                                      2                     1   \n",
       "TAT                                           1                     1   \n",
       "TF                                            1                     1   \n",
       "\n",
       "         gene_to_symptoms_pub_counts gene_edges_out  \n",
       "CDSN                              13            960  \n",
       "TFPI                               1           1275  \n",
       "F5                                 9           1001  \n",
       "SSB                                2           1380  \n",
       "PROS1                             14            705  \n",
       "CD82                               1           1676  \n",
       "KLK3                               2           2156  \n",
       "SERPINE1                           2           1935  \n",
       "SERPINB2                           1           1297  \n",
       "JAK2                               2           2975  \n",
       "TWIST1                             5           3211  \n",
       "TUBB4B                             1           2542  \n",
       "NEUROD1                            1           2477  \n",
       "CP                                 1           2682  \n",
       "GPI                                2           3194  \n",
       "F8                                 7           1148  \n",
       "PLAT                               1           2526  \n",
       "PLG                                1           1275  \n",
       "PROC                               6            821  \n",
       "LPA                                1           1428  \n",
       "AKT1                               0          16400  \n",
       "TGFB1                              4           9079  \n",
       "EGFR                               1           9004  \n",
       "MMP2                               2           7392  \n",
       "MTHFR                              5           1639  \n",
       "IFNA1                              1           7673  \n",
       "PTH                                4           4705  \n",
       "VWF                                1           1607  \n",
       "CISH                               1           6348  \n",
       "ACE                                1           2842  \n",
       "GRK2                               1           4431  \n",
       "KIT                                0           4843  \n",
       "PTEN                              16           5807  \n",
       "ADAMTS13                           2            760  \n",
       "TAT                                1           2926  \n",
       "TF                                 1           2821  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df = pd.DataFrame(final_dict).transpose()\n",
    "final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df.to_csv(disease_csv_file, index = True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 9. Look at pathways and biological processes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# pathways\n",
    "gene_to_pathways = predict_many(relevant_gene_inputs, ['Pathway'])\n",
    "gene_to_pathway_results = {}\n",
    "gene_to_pathway_genes = list(gene_to_pathways[\"output_name\"]) # create list of genes\n",
    "gene_to_pathway_genes = list(dict.fromkeys(gene_to_pathway_genes))  # remove duplicates\n",
    "\n",
    "for gene in gene_to_pathway_genes: \n",
    "    gene_to_pathway_results[gene] = {\n",
    "        'pathway_count' : 0,\n",
    "        \"genes_related\" : []\n",
    "    }\n",
    "\n",
    "for index, row in gene_to_pathways.iterrows():\n",
    "    gene_to_pathway_results[row['output_name']]['pathway_count'] = gene_to_pathway_results[row['output_name']]['pathway_count'] + 1\n",
    "    gene_to_pathway_results[row['output_name']]['genes_related'].append(row['input'])\n",
    "    \n",
    "\n",
    "gene_to_pathway_results = dict(sorted(gene_to_pathway_results.items(), key = lambda x: x[1]['pathway_count'], reverse = True))\n",
    "\n",
    "    \n",
    "gene_to_pathway_results\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_to_bioprocess = predict_many(relevant_gene_inputs, ['BiologicalProcess'])\n",
    "gene_to_bioprocess_results = {}\n",
    "gene_to_bioprocess_genes = list(gene_to_bioprocess[\"output_name\"]) # create list of genes\n",
    "gene_to_bioprocess_genes = list(dict.fromkeys(gene_to_bioprocess_genes))  # remove duplicates\n",
    "\n",
    "for gene in gene_to_bioprocess_genes: \n",
    "    gene_to_bioprocess_results[gene] = {\n",
    "        'bioprocess_count' : 0,\n",
    "        \"genes_related\" : []\n",
    "    }\n",
    "\n",
    "for index, row in gene_to_bioprocess.iterrows():\n",
    "    gene_to_bioprocess_results[row['output_name']]['bioprocess_count'] = gene_to_bioprocess_results[row['output_name']]['bioprocess_count'] + 1\n",
    "    gene_to_bioprocess_results[row['output_name']]['genes_related'].append(row['input'])\n",
    "    \n",
    "gene_to_bioprocess_results = dict(sorted(gene_to_bioprocess_results.items(), key = lambda x: x[1]['bioprocess_count'], reverse = True))\n",
    "\n",
    "for key in gene_to_bioprocess_results.keys(): \n",
    "    if(('C0' in key) or ('C1' in key)): \n",
    "        try: \n",
    "            name = ht.query(key)['BiologicalProcess'][0]['name']\n",
    "            gene_to_bioprocess_results[name] = gene_to_bioprocess_results[key]\n",
    "            del gene_to_bioprocess_results[key]\n",
    "        except: \n",
    "            pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_to_pathway_df = pd.DataFrame(gene_to_pathway_results).transpose()\n",
    "gene_to_bioprocess_df = pd.DataFrame(gene_to_bioprocess_results).transpose()\n",
    "gene_to_pathway_df.to_csv('Dravet_Syndrome_gene_to_pathways_2020_09_14.csv', index = True)\n",
    "gene_to_bioprocess_df.to_csv('Dravet_Syndrome_gene_to_bioprocesses_2020_09_14.csv', index = True)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
